Index: gst-egl/config.h.in
===================================================================
--- gst-egl.orig/config.h.in
+++ gst-egl/config.h.in
@@ -226,8 +226,7 @@
 /* gettext locale dir */
 #undef LOCALEDIR
 
-/* Define to the sub-directory in which libtool stores uninstalled libraries.
-   */
+/* Define to the sub-directory where libtool stores uninstalled libraries. */
 #undef LT_OBJDIR
 
 /* Name of package */
@@ -275,6 +274,9 @@
 /* the target CPU */
 #undef TARGET_CPU
 
+/* Use Wayland EGL window system */
+#undef USE_EGL_WAYLAND
+
 /* Use X11 EGL window system */
 #undef USE_EGL_X11
 
Index: gst-egl/configure.ac
===================================================================
--- gst-egl.orig/configure.ac
+++ gst-egl/configure.ac
@@ -329,6 +329,11 @@ AC_SUBST(X11_CFLAGS)
 AC_SUBST(HAVE_X11)
 AM_CONDITIONAL(HAVE_X11, test "x$with_x11" = "xyes")
 
+AC_ARG_WITH([wayland],
+	AS_HELP_STRING([--with-wayland],[Wayland support]),
+	[with_wayland=yes])
+AS_IF([test "x$with_wayland" = "xyes"], [PKG_CHECK_MODULES(WAYLEGL, wayland-egl, with_wayland=yes, with_wayland=no)])
+
 dnl *** gst-libs/gst/egl ***
 AC_ARG_WITH([egl-window-system],
               AS_HELP_STRING([--with-egl-window-system],[EGL window system to use (auto)]),
@@ -380,6 +385,13 @@ case "$EGL_WINDOW_SYSTEM" in
         EGL_LIBS="$EGL_LIBS $X11_LIBS"
       fi
     fi
+
+    dnl Wayland checks
+    if test "x$HAVE_EGL" = "xyes" -a "x$with_wayland" = "xyes"; then
+       AC_DEFINE(USE_EGL_WAYLAND, [1], [Use Wayland EGL window system])
+       EGL_CFLAGS="$EGL_CFLAGS $WAYLEGL_CFLAGS"
+       EGL_LIBS="$EGL_LIBS $WAYLEGL_LIBS"
+    fi
     ;;
   none|no)
     HAVE_EGL="no"
