From ae048223ed692de173c6a80b2f82cd7459856948 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 13 Oct 2024 09:52:32 -0700
Subject: [PATCH] Makefile fixups for OE builds

Upstream-Status: Inappropriate [embedded specific]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 Makefile.public | 26 +++++++-------------------
 1 file changed, 7 insertions(+), 19 deletions(-)

Index: gst-nvcustomhelper/Makefile.public
===================================================================
--- gst-nvcustomhelper.orig/Makefile.public
+++ gst-nvcustomhelper/Makefile.public
@@ -21,8 +21,8 @@
 # DEALINGS IN THE SOFTWARE.
 ################################################################################ 
 
-CXX:= gcc
 SRCS:= gst-nvcustomevent.c
+libdir ?= /usr/lib
 
 INCS:= $(wildcard *.h)
 OUT_DIR ?= .
@@ -30,38 +30,26 @@ SO_NAME := libgstnvcustomhelper.so
 LIB:= $(OUT_DIR)/$(SO_NAME)
 OBJS :=  $(SRCS:%.c=$(OUT_DIR)/%.o)
 
-TARGET_DEVICE = $(shell gcc -dumpmachine | cut -f1 -d -)
-ifeq ($(TARGET_DEVICE),aarch64)
-  LIB_INSTALL_DIR?=/usr/lib/aarch64-linux-gnu/tegra/
-  CFLAGS:=
-else
-  NVDS_VERSION ?=8.0
-  LIB_INSTALL_DIR ?=/opt/nvidia/deepstream/deepstream-$(NVDS_VERSION)/lib/
-  CFLAGS:= -DDS_VERSION=\"8.0.0\"
-endif
-
 CFLAGS+= -fPIC
 
 LIBS := -shared -Wl,-no-undefined
 
-LIBS+= -L$(LIB_INSTALL_DIR) -ldl\
-        -Wl,-rpath,$(LIB_INSTALL_DIR)
-
-PKGS:= gstreamer-1.0 gstreamer-base-1.0 gstreamer-video-1.0
+PKGS:= gstreamer-1.0
 CFLAGS+= $(shell pkg-config --cflags $(PKGS))
 LIBS+= $(shell pkg-config --libs $(PKGS))
 
 all: $(LIB)
 
 $(OUT_DIR)/%.o: %.c $(INCS) Makefile
-	@echo $(CFLAGS)
-	$(CXX) -c -o $@ $(CFLAGS) $<
+	$(CC) -c -o $@ $(CFLAGS) $<
 
 $(LIB): $(OBJS) Makefile
-	$(CXX) -o $@ $(OBJS) $(LIBS)
+	$(CC) -o $@ $(LDFLAGS) $(OBJS) $(LIBS)
 
 install: $(LIB)
-	cp -rv $(LIB) $(LIB_INSTALL_DIR)
+	install -d $(DESTDIR)$(libdir)
+	install -m0644 $(LIB) $(DESTDIR)$(libdir)/$(SO_NAME).1.0.0
+	ln -sf $(SO_NAME).1.0.0 $(DESTDIR)$(libdir)/$(SO_NAME)
 
 clean:
 	rm -rf $(OBJS) $(LIB)
