From 5ddd035b2f652a5d09248c7e67a1f74683310305 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Thu, 16 Jul 2020 07:54:26 -0700
Subject: [PATCH] Build fixups

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>

---
 Makefile | 28 ++++++++++++----------------
 1 file changed, 12 insertions(+), 16 deletions(-)

Index: gst-nvarguscamera/Makefile
===================================================================
--- gst-nvarguscamera.orig/Makefile
+++ gst-nvarguscamera/Makefile
@@ -27,20 +27,16 @@
 SO_NAME := libgstnvarguscamerasrc.so
 
 CC := g++
+exec_prefix ?= /usr
+libdir ?= $(exec_prefix)/lib
+includedir ?= $(exec_prefix)/include
+
+GST_INSTALL_DIR ?= $(libdir)/gstreamer-1.0
 
-GST_INSTALL_DIR?=/usr/lib/aarch64-linux-gnu/gstreamer-1.0/
-LIB_INSTALL_DIR?=/usr/lib/aarch64-linux-gnu/tegra/
-CFLAGS:=
 LIBS:= -lnvbufsurface -lnvbufsurftransform -lnvdsbufferpool -lnvargus_socketclient -lpthread
 
 SRCS := $(wildcard *.cpp)
 
-INCLUDES += -I./ -I../
-
-# Include jetson_mm_api include path
-INCLUDES += -I/usr/src/jetson_multimedia_api/include/
-INCLUDES += -I/usr/src/jetson_multimedia_api/argus/samples/utils/
-
 PKGS := gstreamer-1.0 \
 	gstreamer-base-1.0 \
 	gstreamer-video-1.0 \
@@ -49,25 +45,26 @@ PKGS := gstreamer-1.0 \
 
 OBJS := $(SRCS:.cpp=.o)
 
-CFLAGS += -fPIC
+CXXFLAGS += -fPIC -I=$(includedir)/Argus/utils
 
-CFLAGS += `pkg-config --cflags $(PKGS)`
+CXXFLAGS += `pkg-config --cflags $(PKGS)`
 
-LDFLAGS = -Wl,--no-undefined -L$(LIB_INSTALL_DIR) -Wl,-rpath,$(LIB_INSTALL_DIR)
+LDFLAGS += -Wl,--no-undefined
 
 LIBS += `pkg-config --libs $(PKGS)`
 
 all: $(SO_NAME)
 
 %.o: %.cpp
-	$(CC) -c $< $(CFLAGS) $(INCLUDES) -o $@
+	$(CXX) $(CXXFLAGS) -c $< -o $@
 
 $(SO_NAME): $(OBJS)
-	$(CC) -shared -o $(SO_NAME) $(OBJS) $(LIBS) $(LDFLAGS)
+	$(CXX) -shared -o $(SO_NAME) $(OBJS) $(LIBS) $(LDFLAGS)
 
 .PHONY: install
 install: $(SO_NAME)
-	cp -vp $(SO_NAME) $(GST_INSTALL_DIR)
+	install -d $(DESTDIR)$(GST_INSTALL_DIR)
+	install -m 0644 $(SO_NAME) $(DESTDIR)$(GST_INSTALL_DIR)/
 
 .PHONY: clean
 clean:
