Index: gst-egl/configure.ac
===================================================================
--- gst-egl.orig/configure.ac
+++ gst-egl/configure.ac
@@ -242,23 +242,6 @@ PKG_CHECK_MODULES(GMODULE_EXPORT, gmodul
 dnl Needed by plugins that use g_module_*() API
 PKG_CHECK_MODULES(GMODULE_NO_EXPORT, gmodule-no-export-2.0)
 
-dnl x11 is optional for librfb and eglglessink
-HAVE_X11=NO
-PKG_CHECK_MODULES(X11, x11, HAVE_X11=yes, HAVE_X11=no)
-AC_SUBST(X11_LIBS)
-AC_SUBST(X11_CFLAGS)
-AC_SUBST(HAVE_X11)
-AM_CONDITIONAL(HAVE_X11, test "x$HAVE_X11" = "xyes")
-if test "x$HAVE_X11" = "xyes"; then
-  AC_DEFINE(HAVE_X11, 1, [Define if you have X11 library])
-fi
-
-dnl exif (used on jifmux tests) ****
-PKG_CHECK_MODULES(EXIF, libexif >= 0.6.16, HAVE_EXIF="yes", HAVE_EXIF="no")
-AC_SUBST(EXIF_LIBS)
-AC_SUBST(EXIF_CFLAGS)
-AM_CONDITIONAL(USE_EXIF, test "x$HAVE_EXIF" = "xyes")
-
 dnl Orc
 ORC_CHECK([0.4.17])
 
@@ -335,15 +318,26 @@ AG_GST_DEFAULT_ELEMENTS
 
 dnl *** plug-ins to exclude ***
 
+dnl x11 is optional for librfb and eglglessink
+AC_ARG_WITH([x11],,[with_x11=yes])
+
+AS_IF([test "x$with_x11" = "xyes"],[PKG_CHECK_MODULES(X11, x11, with_x11=yes, with_x11=no)])
+AS_IF([test "x$with_x11" = "xyes"], AC_DEFINE(HAVE_X11, 1, [Define if you have X11 library]))
+
+AC_SUBST(X11_LIBS)
+AC_SUBST(X11_CFLAGS)
+AC_SUBST(HAVE_X11)
+AM_CONDITIONAL(HAVE_X11, test "x$with_x11" = "xyes")
+
 dnl *** gst-libs/gst/egl ***
 AC_ARG_WITH([egl-window-system],
-              AS_HELP_STRING([--with-egl-window-system],[EGL window system to use (x11)]),
+              AS_HELP_STRING([--with-egl-window-system],[EGL window system to use (auto)]),
               [EGL_WINDOW_SYSTEM="$withval"],
               [EGL_WINDOW_SYSTEM="auto"])
 
 if test x"$EGL_WINDOW_SYSTEM" = x"auto"; then
   if test x"$EGL_WINDOW_SYSTEM" = x"auto"; then
-      if test x"$HAVE_X11" = x"yes"; then
+      if test x"$with_x11" = x"yes"; then
         EGL_WINDOW_SYSTEM="x11"
       fi
   fi
@@ -378,7 +372,7 @@ case "$EGL_WINDOW_SYSTEM" in
       
     dnl X11 specific part, above is auto and X11
     if test x"$HAVE_EGL" = x"yes" -a x"$EGL_WINDOW_SYSTEM" = x"x11"; then
-      if test x"$HAVE_X11" != x"yes"; then
+      if test x"$with_x11" != x"yes"; then
         AC_MSG_ERROR([libX11 not found and is required for EGL X11 window system])
       else
         AC_DEFINE(USE_EGL_X11, [1], [Use X11 EGL window system])
Index: gst-egl/ext/eglgles/gsteglglessink.c
===================================================================
--- gst-egl.orig/ext/eglgles/gsteglglessink.c
+++ gst-egl/ext/eglgles/gsteglglessink.c
@@ -107,7 +107,9 @@
 #include <gst/video/gstvideopool.h>
 #include <gst/video/videooverlay.h>
 
+#ifdef USE_EGL_X11
 #include <X11/Xlib.h>
+#endif
 
 #include "gstegladaptation.h"
 #include "video_platform_wrapper.h"
