From b56aaaa3f571d07c172ee5a12e0bd34a5d12ca4f Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sat, 30 Mar 2019 08:35:01 -0700
Subject: [PATCH 8/9] libweston: have compositor-drm try drmOpen should udev
 search fail

Signed-off-by: Matt Madison <matt@madison.systems>
---
 libweston/compositor-drm.c | 36 ++++++++++++++++++++++++++++--------
 1 file changed, 28 insertions(+), 8 deletions(-)

diff --git a/libweston/compositor-drm.c b/libweston/compositor-drm.c
index 100b3050..336cc7a2 100644
--- a/libweston/compositor-drm.c
+++ b/libweston/compositor-drm.c
@@ -399,6 +399,7 @@ struct drm_fb {
 
 	/* Used by dumb fbs */
 	void *map;
+	bool nounmap;
 };
 
 struct drm_edid {
@@ -969,7 +970,7 @@ drm_fb_destroy_dumb(struct drm_fb *fb)
 
 	assert(fb->type == BUFFER_PIXMAN_DUMB);
 
-	if (fb->map && fb->size > 0)
+	if (fb->map && fb->size > 0 && !fb->nounmap)
 		munmap(fb->map, fb->size);
 
 	memset(&destroy_arg, 0, sizeof(destroy_arg));
@@ -1095,10 +1096,16 @@ drm_fb_create_dumb(struct drm_backend *b, int width, int height,
 	if (ret)
 		goto err_add_fb;
 
-	fb->map = mmap(NULL, fb->size, PROT_WRITE,
-		       MAP_SHARED, b->drm.fd, map_arg.offset);
-	if (fb->map == MAP_FAILED)
-		goto err_add_fb;
+	if (!strcmp(b->drm.filename, "drm-nvdc")) {
+		fb->map = map_arg.offset;
+		fb->nounmap = true;
+	} else {
+		fb->map = mmap(NULL, fb->size, PROT_WRITE,
+			       MAP_SHARED, b->drm.fd, map_arg.offset);
+		if (fb->map == MAP_FAILED)
+			goto err_add_fb;
+		fb->nounmap = false;
+	}
 
 	return fb;
 
@@ -7607,8 +7614,20 @@ drm_backend_create(struct weston_compositor *compositor,
 	else
 		drm_device = find_primary_gpu(b, seat_id);
 	if (drm_device == NULL) {
-		weston_log("no drm device found\n");
-		goto err_udev;
+		int fd = drmOpen("drm-nvdc", "");
+		char *name;
+		if (fd >= 0) {
+			name = drmGetDeviceNameFromFd(fd);
+			b->drm.fd = fd;
+			b->drm.id = -1;
+			if (name) {
+				b->drm.filename = strdup(name);
+				drmFree(name);
+			}
+		} else {
+			weston_log("no drm device found\n");
+			goto err_udev;
+		}
 	}
 
 	if (init_kms_caps(b) < 0) {
@@ -7678,7 +7697,8 @@ drm_backend_create(struct weston_compositor *compositor,
 		goto err_udev_monitor;
 	}
 
-	udev_device_unref(drm_device);
+	if (drm_device)
+		udev_device_unref(drm_device);
 
 	weston_compositor_add_debug_binding(compositor, KEY_O,
 					    planes_binding, b);
-- 
2.19.1

