Index: cmake-3.14.1/Modules/FindCUDA.cmake
===================================================================
--- cmake-3.14.1.orig/Modules/FindCUDA.cmake
+++ cmake-3.14.1/Modules/FindCUDA.cmake
@@ -877,14 +877,7 @@ if(CUDA_USE_STATIC_CUDA_RUNTIME)
   if(UNIX)
     # Check for the dependent libraries.
 
-    # Many of the FindXYZ CMake comes with makes use of try_compile with int main(){return 0;}
-    # as the source file.  Unfortunately this causes a warning with -Wstrict-prototypes and
-    # -Werror causes the try_compile to fail.  We will just temporarily disable other flags
-    # when doing the find_package command here.
-    set(_cuda_cmake_c_flags ${CMAKE_C_FLAGS})
-    set(CMAKE_C_FLAGS "-fPIC")
     find_package(Threads REQUIRED)
-    set(CMAKE_C_FLAGS ${_cuda_cmake_c_flags})
 
     if(NOT APPLE)
       #On Linux, you must link against librt when using the static cuda runtime.
