Index: cmake-3.4.3/Modules/FindCUDA.cmake
===================================================================
--- cmake-3.4.3.orig/Modules/FindCUDA.cmake
+++ cmake-3.4.3/Modules/FindCUDA.cmake
@@ -746,14 +746,7 @@ if(CUDA_USE_STATIC_CUDA_RUNTIME)
     endif()
     set(CMAKE_THREAD_PREFER_PTHREAD 1)
 
-    # Many of the FindXYZ CMake comes with makes use of try_compile with int main(){return 0;}
-    # as the source file.  Unfortunately this causes a warning with -Wstrict-prototypes and
-    # -Werror causes the try_compile to fail.  We will just temporarily disable other flags
-    # when doing the find_package command here.
-    set(_cuda_cmake_c_flags ${CMAKE_C_FLAGS})
-    set(CMAKE_C_FLAGS "-fPIC")
     find_package(Threads REQUIRED)
-    set(CMAKE_C_FLAGS ${_cuda_cmake_c_flags})
 
     if (DEFINED _cuda_cmake_thread_prefer_pthread)
       set(CMAKE_THREAD_PREFER_PTHREAD ${_cuda_cmake_thread_prefer_pthread})
