From 3493b79dbf37a5d48c98c328a679ae484e6c3a8a Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Mon, 28 Dec 2020 05:59:33 -0800
Subject: [PATCH 10/11] t194: add bootinfo to build

Signed-off-by: Matt Madison <matt@madison.systems>
---
 bootloader/partner/t18x/cboot/platform/t194/rules.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bootloader/partner/t18x/cboot/platform/t194/rules.mk b/bootloader/partner/t18x/cboot/platform/t194/rules.mk
index 2ce07f9..e6e7e7b 100644
--- a/bootloader/partner/t18x/cboot/platform/t194/rules.mk
+++ b/bootloader/partner/t18x/cboot/platform/t194/rules.mk
@@ -77,6 +77,7 @@ MODULE_DEPS += \
 	$(LOCAL_DIR)/../../../../common/drivers/pwm \
 	$(LOCAL_DIR)/../../../../common/drivers/display \
 	$(LOCAL_DIR)/../../../../common/lib/cbo \
+	$(LOCAL_DIR)/../../../../common/lib/bootinfo \
 	$(LOCAL_DIR)/../../../../$(TARGET_FAMILY)/common/lib/device_prod
 
 ifeq ($(filter t19x, $(TARGET_FAMILY)),)
-- 
2.30.2

