From eee11e89e371b6c9eb9a57611c05af0c0da33d91 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ilies.chergui@gmail.com>
Date: Thu, 21 Apr 2022 12:56:20 +0100
Subject: [PATCH] arm64: set -mno-outline-atomics and -fno-stack-protector in
 CFLAGS

Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>
---
 utils.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/utils.mk b/utils.mk
index 2b15880..6c8808c 100644
--- a/utils.mk
+++ b/utils.mk
@@ -62,6 +62,8 @@ HOST_CFLAGS           += -Wno-unused-parameter
 # Treat warnings as errors, if requested
 WARNINGS_AS_ERRORS    ?=
 CFLAGS                += $(if $(WARNINGS_AS_ERRORS),-Werror)
+CFLAGS                += -mno-outline-atomics
+CFLAGS                += -fno-stack-protector
 
 DEBUG                 ?=
 DEVELOP               ?=
-- 
2.32.0

