From b025626b42f892bfe1f3683531a9031fec4f6834 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Fri, 23 Jul 2021 11:04:56 -0700
Subject: [PATCH] Update makefile for statically linking external libtirpc

Signed-off-by: Matt Madison <matt@madison.systems>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index cd44c73..6f24b80 100644
--- a/Makefile
+++ b/Makefile
@@ -127,9 +127,9 @@ LDLIBS   := $(LDLIBS)
 # Library flags (recursively expanded to handle target-specific flags)
 LIB_CPPFLAGS       = -DNV_LINUX -isystem $(DEPS_DIR)$(includedir) -include $(BUILD_DEFS)
 LIB_CFLAGS         = -fPIC
-LIB_LDFLAGS        = -L$(DEPS_DIR)$(libdir) -shared -Wl,-soname=$(LIB_SONAME)
-LIB_LDLIBS_STATIC  = -l:libnvidia-modprobe-utils.a
-LIB_LDLIBS_SHARED  = -ldl -lcap -ltirpc
+LIB_LDFLAGS        = -shared -Wl,-soname=$(LIB_SONAME)
+LIB_LDLIBS_STATIC  = -L$(DEPS_DIR)$(libdir) -l:libnvidia-modprobe-utils.a -L=$(libdir) -l:libtirpc.a
+LIB_LDLIBS_SHARED  = -ldl -lcap -lpthread
 ifeq ($(WITH_LIBELF), yes)
 LIB_CPPFLAGS       += -DWITH_LIBELF
 LIB_LDLIBS_SHARED  += -lelf
-- 
2.30.2

