From 719ceba2aa4762e80902235666cbf6f02c8f731d Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Mon, 18 May 2026 14:20:29 -0700
Subject: [PATCH] include: fix jpeglib header inclusion

To ensure that it's using the NVIDIA-specific header.

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 include/NvJpegDecoder.h | 2 +-
 include/NvJpegEncoder.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/NvJpegDecoder.h b/include/NvJpegDecoder.h
index 9b877fa..454b9a7 100644
--- a/include/NvJpegDecoder.h
+++ b/include/NvJpegDecoder.h
@@ -58,7 +58,7 @@
 #endif
 
 #include <stdio.h>
-#include "jpeglib.h"
+#include "libjpeg-9f/jpeglib.h"
 #include "NvElement.h"
 #include "NvBuffer.h"
 
diff --git a/include/NvJpegEncoder.h b/include/NvJpegEncoder.h
index cb8a763..0be4525 100644
--- a/include/NvJpegEncoder.h
+++ b/include/NvJpegEncoder.h
@@ -59,7 +59,7 @@
 #endif
 
 #include <stdio.h>
-#include "jpeglib.h"
+#include "libjpeg-9f/jpeglib.h"
 #include "NvElement.h"
 #include "NvBuffer.h"
 
-- 
2.43.0

