From 2206062fca84109db3456b8730a09ffaf197ca27 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ilies.chergui@gmail.com>
Date: Sun, 8 Aug 2021 21:18:48 +0100
Subject: [PATCH 2/3] OE cross-build fixups

Signed-off-by: Pablo Rodriguez Quesada <pablo.rodriguez-quesada@windriver.com>
Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>
Signed-off-by: Matt Madison <matt@madison.systems>

Upstream-Status: Pending
---
 Makefile     | 40 ++++++++++++++++++++++------------------
 mk/common.mk |  4 ++--
 2 files changed, 24 insertions(+), 20 deletions(-)

diff --git a/Makefile b/Makefile
index 709102b7..da836fae 100644
--- a/Makefile
+++ b/Makefile
@@ -23,17 +23,17 @@ WITH_SECCOMP ?= yes
 
 ##### Global definitions #####
 
-export prefix      = /usr/local
-export exec_prefix = $(prefix)
-export libdir      = $(exec_prefix)/lib
-export docdir      = $(prefix)/share/doc
-export libdbgdir   = $(prefix)/lib/debug$(libdir)
-export includedir  = $(prefix)/include
-export pkgconfdir  = $(libdir)/pkgconfig
+export prefix      ?= /usr/local
+export exec_prefix ?= $(prefix)
+export libdir      ?= $(exec_prefix)/lib
+export docdir      ?= $(prefix)/share/doc
+export libdbgdir   ?= $(prefix)/lib/debug$(libdir)
+export includedir  ?= $(prefix)/include
+export pkgconfdir  ?= $(libdir)/pkgconfig
 
 export PKG_DIR     ?= $(CURDIR)/pkg
 export TESTS_DIR   ?= $(CURDIR)/test
-export SRCS_DIR    ?= $(CURDIR)/src
+export SRCS_DIR    ?= src
 export DEPS_DIR    ?= $(CURDIR)/deps
 export DIST_DIR    ?= $(CURDIR)/dist
 export MAKE_DIR    ?= $(CURDIR)/mk
@@ -119,21 +119,23 @@ LIB_PKGCFG  := $(LIB_NAME).pc
 ##### Flags definitions #####
 
 # Common flags
-CPPFLAGS := -D_GNU_SOURCE -D_FORTIFY_SOURCE=2 -g3 -D JETSON=$(JETSON) $(CPPFLAGS)
+CPPFLAGS := -D_GNU_SOURCE -D_FORTIFY_SOURCE=2 -g3 -D JETSON=$(JETSON) -DHOST_LIBDIR=$(libdir) $(CPPFLAGS)
+CCNAME := $(shell $(CC) -v 2>&1 | grep -q "clang version" && echo clang || echo gcc)
 CFLAGS   := -std=gnu11 -O0 -g3 -fdata-sections -ffunction-sections -fstack-protector -fno-strict-aliasing -fvisibility=hidden \
             -Wall -Wextra -Wcast-align -Wpointer-arith -Wmissing-prototypes -Wnonnull \
             -Wwrite-strings -Wlogical-op -Wformat=2 -Wmissing-format-attribute -Winit-self -Wshadow \
             -Wstrict-prototypes -Wunreachable-code -Wconversion -Wsign-conversion \
-            -Wno-unknown-warning-option -Wno-format-extra-args -Wno-gnu-alignof-expression $(CFLAGS)
+            -Wno-format-extra-args $(if $(filter clang,$(CCNAME)),-Wno-unknown-warning-option -Wno-gnu-alignof-expression,) \
+            -I=/usr/include/tirpc $(CFLAGS)
 LDFLAGS  := -Wl,-zrelro -Wl,-znow -Wl,-zdefs -Wl,--gc-sections $(LDFLAGS)
 LDLIBS   := $(LDLIBS)
 
 # Library flags (recursively expanded to handle target-specific flags)
 LIB_CPPFLAGS       = -DNV_LINUX -isystem $(DEPS_DIR)$(includedir) -include $(BUILD_DEFS)
 LIB_CFLAGS         = -fPIC
-LIB_LDFLAGS        = -L$(DEPS_DIR)$(libdir) -shared -Wl,-soname=$(LIB_SONAME)
-LIB_LDLIBS_STATIC  = -l:libnvidia-modprobe-utils.a
-LIB_LDLIBS_SHARED  = -ldl -lcap
+LIB_LDFLAGS        = -shared -Wl,-soname=$(LIB_SONAME)
+LIB_LDLIBS_STATIC  = -L$(DEPS_DIR)$(libdir) -l:libnvidia-modprobe-utils.a
+LIB_LDLIBS_SHARED  = -ltirpc -ldl -lcap -lpthread
 ifeq ($(WITH_LIBELF), yes)
 LIB_CPPFLAGS       += -DWITH_LIBELF
 LIB_LDLIBS_SHARED  += -lelf
@@ -170,13 +172,17 @@ DEPENDENCIES   := $(LIB_OBJS:%.lo=%.d)
 $(BUILD_DEFS):
 	@printf '#define BUILD_DATE     "%s"\n' '$(strip $(DATE))' >$(BUILD_DEFS)
 	@printf '#define BUILD_COMPILER "%s " __VERSION__\n' '$(notdir $(COMPILER))' >>$(BUILD_DEFS)
+ifeq ($(EXCLUDE_BUILD_FLAGS),)
 	@printf '#define BUILD_FLAGS    "%s"\n' '$(strip $(CPPFLAGS) $(CFLAGS) $(LDFLAGS))' >>$(BUILD_DEFS)
+else
+	@printf '#define BUILD_FLAGS    ""\n' >>$(BUILD_DEFS)
+endif
 	@printf '#define BUILD_REVISION "%s"\n' '$(strip $(REVISION))' >>$(BUILD_DEFS)
 	@printf '#define BUILD_PLATFORM "%s"\n' '$(strip $(PLATFORM))' >>$(BUILD_DEFS)
 
 $(LIB_RPC_SRCS): $(LIB_RPC_SPEC)
 	$(RM) $@
-	cd $(dir $@) && $(RPCGEN) $(RPCGENFLAGS) -C -M -N -o $(notdir $@) $(LIB_RPC_SPEC)
+	cd $(dir $@) && $(RPCGEN) $(RPCGENFLAGS) -C -M -N -o $(notdir $@) $(notdir $(LIB_RPC_SPEC))
 
 $(LIB_OBJS): %.lo: %.c | deps
 	$(CC) $(LIB_CFLAGS) $(LIB_CPPFLAGS) -MMD -MF $*.d -c $(OUTPUT_OPTION) $<
@@ -189,24 +195,22 @@ $(LIB_SHARED): $(LIB_OBJS)
 	$(OBJCPY) --only-keep-debug $@ $(LIB_SONAME)
 	$(OBJCPY) --add-gnu-debuglink=$(LIB_SONAME) $@
 	$(MV) $(LIB_SONAME) $(DEBUG_DIR)
-	$(STRIP) --strip-unneeded -R .comment $@
 
 $(LIB_STATIC_OBJ): $(LIB_OBJS)
 	# FIXME Handle user-defined LDFLAGS and LDLIBS
 	$(LD) -d -r --exclude-libs ALL -L$(DEPS_DIR)$(libdir) $(OUTPUT_OPTION) $^ $(LIB_LDLIBS_STATIC)
 	$(OBJCPY) --localize-hidden $@
-	$(STRIP) --strip-unneeded -R .comment $@
 
 ##### Public rules #####
 
 all: CPPFLAGS += -DNDEBUG
-all: STRIP  := @echo skipping: strip
+all:
 all: shared static
 
 # Run with ASAN_OPTIONS="protect_shadow_gap=0" to avoid CUDA OOM errors
 debug: CFLAGS += -pedantic -fsanitize=undefined -fno-omit-frame-pointer -fno-common -fsanitize=address
 debug: LDLIBS += -lubsan
-debug: STRIP  := @echo skipping: strip
+debug:
 debug: shared static
 
 shared: $(LIB_SHARED)
diff --git a/mk/common.mk b/mk/common.mk
index d6ef499d..f8170de8 100644
--- a/mk/common.mk
+++ b/mk/common.mk
@@ -22,9 +22,9 @@ UID      := $(shell id -u)
 GID      := $(shell id -g)
 DATE     := $(shell date -u --iso-8601=minutes)
 REVISION := $(shell git rev-parse HEAD)
-COMPILER := $(realpath $(shell which $(CC)))
+COMPILER := $(realpath $(shell which $(firstword $(CC))))
 PLATFORM ?= $(shell uname -m)
-JETSON   := TRUE
+JETSON   ?= TRUE
 
 ifeq ($(DATE),)
 $(error Invalid date format)
-- 
2.34.1

