Index: gst-omx1/omx/Makefile.am
===================================================================
--- gst-omx1.orig/omx/Makefile.am
+++ gst-omx1/omx/Makefile.am
@@ -19,12 +19,14 @@ libgstomx_la_SOURCES = \
 	gstomxmjpegdec.c \
 	gstomxmpeg4videodec.c \
 	gstomxmpeg2videodec.c \
+	gstomxh265dec.c \
 	gstomxh264dec.c \
 	gstomxh263dec.c \
 	gstomxwmvdec.c \
 	$(VP8_C_FILES) \
 	$(THEORA_C_FILES) \
 	gstomxmpeg4videoenc.c \
+	gstomxh265enc.c \
 	gstomxh264enc.c \
 	gstomxh263enc.c \
 	gstomxaacenc.c
@@ -50,12 +52,14 @@ noinst_HEADERS = \
 	gstomxmjpegdec.h \
 	gstomxmpeg2videodec.h \
 	gstomxmpeg4videodec.h \
+	gstomxh265dec.h \
 	gstomxh264dec.h \
 	gstomxh263dec.h \
 	gstomxwmvdec.h \
 	$(VP8_H_FILES) \
 	$(THEORA_H_FILES) \
 	gstomxmpeg4videoenc.h \
+	gstomxh265enc.h \
 	gstomxh264enc.h \
 	gstomxh263enc.h \
 	gstomxaacenc.h
