From 55e1503e104a9e9bd275d683146db0833b7301f4 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Tue, 22 Oct 2024 13:02:51 -0700
Subject: [PATCH] Update makefile for OE builds

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 Makefile | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

Index: gst-nvunixfd/Makefile
===================================================================
--- gst-nvunixfd.orig/Makefile
+++ gst-nvunixfd/Makefile
@@ -17,25 +17,11 @@ endif
 
 SO_NAME := libgstnvunixfd.so
 
-CC := gcc
-
-TARGET_DEVICE = $(shell gcc -dumpmachine | cut -f1 -d -)
-ifeq ($(TARGET_DEVICE),aarch64)
-	GST_INSTALL_DIR?=/usr/lib/aarch64-linux-gnu/gstreamer-1.0/
-	LIB_INSTALL_DIR?=/opt/nvidia/deepstream/deepstream/lib/
-else
-	GST_INSTALL_DIR?=/usr/lib/x86_64-linux-gnu/gstreamer-1.0/
-	LIB_INSTALL_DIR?=/opt/nvidia/deepstream/deepstream/lib/
-endif
-
-CUDA_HOME:= /usr/local/cuda-$(CUDA_VER)
-CFLAGS:=
 LIBS:= -lnvbufsurface -lnvbufsurftransform -lpthread
 
 SRCS := $(wildcard *.c)
 
 INCLUDES += -I./ -I../
-INCLUDES += -I$(CUDA_HOME)/include
 
 PKGS := gstreamer-1.0 \
 	gstreamer-base-1.0 \
@@ -43,7 +29,8 @@ PKGS := gstreamer-1.0 \
 	gstreamer-allocators-1.0 \
 	glib-2.0 \
 	gio-2.0 \
-	gio-unix-2.0
+	gio-unix-2.0 \
+	cuda-$(CUDA_VER) cudart-$(CUDA_VER)
 
 OBJS := $(SRCS:.c=.o)
 
@@ -54,12 +41,10 @@ CFLAGS += -fPIC \
 
 CFLAGS += `pkg-config --cflags $(PKGS)`
 
-LDFLAGS = -Wl,--no-undefined -L$(LIB_INSTALL_DIR) -Wl,-rpath,$(LIB_INSTALL_DIR)
+LDFLAGS += -Wl,--no-undefined
 
 LIBS += `pkg-config --libs $(PKGS)`
 
-LIBS += -L$(CUDA_HOME)/lib64 -lcuda -lcudart
-
 all: $(SO_NAME)
 
 %.o: %.c
@@ -71,7 +56,7 @@ $(SO_NAME): $(OBJS)
 
 .PHONY: install
 install: $(SO_NAME)
-	cp -vp $(SO_NAME) $(GST_INSTALL_DIR)
+	install -D -m 0644 $(SO_NAME) -t $(DESTDIR)$(libdir)/gstreamer-1.0
 
 .PHONY: clean
 clean:
