From 9dd8eef6123e281112f883c7539475e88ab48c5d Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Fri, 17 Mar 2023 06:47:40 -0700
Subject: [PATCH 2/2] Fix broken override of CUDA_TOOLKIT_TARGET_DIR setting

Upstream-Status: Inappropriate [oe specific]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 cmake/OpenCVDetectCUDA.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/OpenCVDetectCUDA.cmake b/cmake/OpenCVDetectCUDA.cmake
index e0c539b90a..b49bd2e485 100644
--- a/cmake/OpenCVDetectCUDA.cmake
+++ b/cmake/OpenCVDetectCUDA.cmake
@@ -11,7 +11,9 @@ endif()
 #set(OPENCV_CMAKE_CUDA_DEBUG 1)
 
 if(CUDA_TOOLKIT_ROOT_DIR)
-  set(CUDA_TOOLKIT_TARGET_DIR ${CUDA_TOOLKIT_ROOT_DIR})
+  if(NOT CUDA_TOOLKIT_TARGET_DIR)
+    set(CUDA_TOOLKIT_TARGET_DIR ${CUDA_TOOLKIT_ROOT_DIR})
+  endif()
 endif()
 
 if(((NOT CMAKE_VERSION VERSION_LESS "3.9.0")  # requires https://gitlab.kitware.com/cmake/cmake/merge_requests/663
@@ -245,8 +247,6 @@ if(HAVE_CUDA)
   foreach(p ${CUDA_LIBS_PATH})
     if(MSVC AND CMAKE_GENERATOR MATCHES "Ninja|JOM")
       set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} ${CMAKE_LIBRARY_PATH_FLAG}"${p}")
-    else()
-      set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} ${CMAKE_LIBRARY_PATH_FLAG}${p})
     endif()
   endforeach()
 endif()
-- 
2.34.1

