From 00529c2b30cf74993bb124ba82a6947703540584 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 17 Mar 2024 09:33:12 -0700
Subject: [PATCH] Fix cgo LDFLAGS for go 1.21 and later

Upstream-Status: Pending

Signed-off-by: Matt Madison <matt@madison.systems>
---
 internal/cuda/cuda.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/internal/cuda/cuda.go b/internal/cuda/cuda.go
index 2c70a82..3fc7952 100644
--- a/internal/cuda/cuda.go
+++ b/internal/cuda/cuda.go
@@ -23,7 +23,7 @@ import (
 )
 
 /*
-#cgo LDFLAGS: -Wl,--unresolved-symbols=ignore-in-object-files
+#cgo LDFLAGS: -Wl,--export-dynamic -Wl,--unresolved-symbols=ignore-in-object-files
 
 #ifdef _WIN32
 #define CUDAAPI __stdcall
-- 
2.34.1

