From e3a1f1087efb25d780c638e32c13ebd5de62b065 Mon Sep 17 00:00:00 2001
From: Kurt Kiefer <kekiefer@gmail.com>
Date: Tue, 15 Oct 2019 10:41:51 -0700
Subject: [PATCH] Enable build without X11 headers

---
 include/EGL/eglplatform.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/EGL/eglplatform.h b/include/EGL/eglplatform.h
index 29ab288..e21fc20 100644
--- a/include/EGL/eglplatform.h
+++ b/include/EGL/eglplatform.h
@@ -118,6 +118,14 @@ typedef intptr_t EGLNativeWindowType;
 
 #elif defined(__unix__) || defined(USE_X11)
 
+#if defined(MESA_EGL_NO_X11_HEADERS)
+
+typedef void            *EGLNativeDisplayType;
+typedef khronos_uintptr_t EGLNativePixmapType;
+typedef khronos_uintptr_t EGLNativeWindowType;
+
+#else
+
 /* X11 (tentative)  */
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
@@ -126,6 +134,8 @@ typedef Display *EGLNativeDisplayType;
 typedef Pixmap   EGLNativePixmapType;
 typedef Window   EGLNativeWindowType;
 
+#endif
+
 #elif defined(__APPLE__)
 
 typedef int   EGLNativeDisplayType;
-- 
2.21.0

