From 5a54920d28ff1b9c26ae4131b2fd0dda5ada9625 Mon Sep 17 00:00:00 2001
From: Daniel Chaves <dchvs11@gmail.com>
Date: Fri, 2 May 2025 23:34:26 +0000
Subject: [PATCH] Drop bc usage and remove symlink creation functions

Signed-off-by: Matt Madison <matt@madison.systems>
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
Signed-off-by: Daniel Chaves <dchvs11@gmail.com>

Upstream-Status: Pending

---
 etc/systemd/nvpower.sh | 154 -----------------------------------------
 1 file changed, 154 deletions(-)

diff --git a/etc/systemd/nvpower.sh b/etc/systemd/nvpower.sh
index 07ff45c..c74208b 100755
--- a/etc/systemd/nvpower.sh
+++ b/etc/systemd/nvpower.sh
@@ -164,158 +164,6 @@ function is_nvpm_relinking_needed()
 	return 1
 }
 
-function create_nvpmodel_symlink()
-{
-	conf_file=""
-
-	if [ -e "${NVPM_SYMLINK}" ]; then
-		if is_nvpm_relinking_needed; then
-			nvpm_back="/etc/nvpmodel_nvbackup.conf"
-			echo "${SCRIPT_NAME} - NOTICE: Backing up the" \
-				"existing ${NVPM_SYMLINK} to ${nvpm_back}!"
-			cp -b -S .bak "${NVPM_SYMLINK}" "${nvpm_back}"
-
-			unlink "${NVPM_SYMLINK}"
-			ret="$?"
-			if [ "${ret}" -ne "0" ]; then
-				echo "${SCRIPT_NAME} - Error: Failed" \
-					"to unlink ${NVPM_SYMLINK}!"
-			else
-				rm /var/lib/nvpmodel/status
-			fi
-		fi
-	fi
-
-	# create /etc/nvpmodel.conf symlink
-	if [ ! -e "${NVPM_SYMLINK}" ]; then
-		if [ "${SOCFAMILY}" = "tegra194" ]; then
-			if [ "${machine}" = "e3900" ]; then
-				if [ -d "/sys/devices/gpu.0" ] &&
-					[ -d "/sys/devices/17000000.gv11b" ]; then
-					conf_file="/etc/nvpmodel/nvpmodel_t194_e3900_iGPU.conf"
-				else
-					conf_file="/etc/nvpmodel/nvpmodel_t194_e3900_dGPU.conf"
-				fi
-			elif [ "${machine}" = "p2972-0006" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_t194_8gb.conf"
-			elif [ "${machine}" = "p3668" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_t194_p3668.conf"
-			elif [ "${machine}" = "p3668-emul" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_t194_p3668_emul.conf"
-			elif [ "${machine}" = "jetson-xavier-industrial" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_t194_agxi.conf"
-			else
-				conf_file="/etc/nvpmodel/nvpmodel_t194.conf"
-			fi
-		elif [ "${SOCFAMILY}" = "tegra234" ]; then
-			if [ "${machine}" = "p3701-0000-as-p3767-0000" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0000.conf"
-			elif [ "${machine}" = "p3701-0000-as-p3767-0001" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0001.conf"
-			elif [ "${machine}" = "p3701-0000-as-p3767-0003" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0003.conf"
-			elif [ "${machine}" = "p3701-0000-as-p3767-0004" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0004.conf"
-			elif [ "${machine}" = "p3701-0000-as-pxxxx" ] || \
-				[ "${machine}" = "e2421-1099-as-pxxxx" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_pxxxx.conf"
-			elif [ "${machine}" = "p3701-0002" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3701_0002.conf"
-			elif [ "${machine}" = "p3701-0000-as-p3701-0004" ] || \
-				[ "${machine}" = "p3701-0004" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3701_0004.conf"
-			elif [ "${machine}" = "p3701-0008" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3701_0008.conf"
-				if [ "${cvb}" = "p3740-0002-c01" ]; then
-					conf_file="/etc/nvpmodel/nvpmodel_p3701_0008_3740_0002.conf"
-				fi
-			elif [ "${machine}" = "p3767-0000" ] || \
-				[ "${machine}" = "p3767-0002" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0000.conf"
-			elif [ "${machine}" = "p3767-0000-super" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0000_super.conf"
-			elif [ "${machine}" = "p3767-0000-px1" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0000_px1.conf"
-			elif [ "${machine}" = "p3767-0000-as-p3767-0001" ] || \
-				[ "${machine}" = "p3767-0001" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0001.conf"
-			elif [ "${machine}" = "p3767-0001-super" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0001_super.conf"
-			elif [ "${machine}" = "p3767-0003" ] || \
-				[ "${machine}" = "p3767-0005" ] || \
-				[ "${machine}" = "p3767-0000-as-p3767-0003" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0003.conf"
-			elif [ "${machine}" = "p3767-0003-super" ] || \
-				[ "${machine}" = "p3767-0005-super" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0003_super.conf"
-			elif [ "${machine}" = "p3767-0004" ] || \
-				[ "${machine}" = "p3767-0000-as-p3767-0004" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0004.conf"
-			elif [ "${machine}" = "p3767-0004-super" ]; then
-				conf_file="/etc/nvpmodel/nvpmodel_p3767_0004_super.conf"
-			else
-				conf_file="/etc/nvpmodel/nvpmodel_p3701_0000.conf"
-			fi
-		fi
-
-		if [ "${conf_file}" != "" ]; then
-			if [ -e "${conf_file}" ]; then
-				ln -sf "${conf_file}" "${NVPM_SYMLINK}"
-			else
-				echo "${SCRIPT_NAME} - WARNING: file ${conf_file} not found!"
-			fi
-		fi
-	fi
-}
-
-function create_nvfancontrol_symlink()
-{
-	conf_file=""
-	if [ ! -e "/etc/nvfancontrol.conf" ]; then
-		if [ "${SOCFAMILY}" = "tegra194" ]; then
-			if [ "${machine}" = "e3900" ]; then
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_e3900.conf"
-			elif [ "${machine}" = "p3668" ]; then
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3668.conf"
-			else
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p2888.conf"
-			fi
-		fi
-
-		if [ "${SOCFAMILY}" = "tegra234" ]; then
-			if [[ "${machine}" =~ "p3701" ]]; then
-				# Use p3701_0000 as default fan settings for p3701 series
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3701_0000.conf"
-				if [ "${machine}" = "p3701-0008" ]; then
-					conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3701_0008.conf"
-				fi
-			elif [[ "${machine}" =~ "p3767" ]]; then
-				# All p3767 series share the same fan settings
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3767_0000.conf"
-			else
-				# Use p3701_0000 as default fan settings for other tegra234 platforms
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3701_0000.conf"
-			fi
-
-			if [ "${cvb}" = "p3711-0000" ]; then
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3711_0000.conf"
-			elif [ "${cvb}" = "p3740-0002-b01" ]; then
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_b01.conf"
-			elif [ "${cvb}" = "p3740-0002-c01" ]; then
-				conf_file="/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_c01.conf"
-			fi
-		fi
-
-		if [ "${conf_file}" != "" ]; then
-			if [ -e "${conf_file}" ]; then
-				ln -sf "${conf_file}" /etc/nvfancontrol.conf
-			else
-				echo "${SCRIPT_NAME} - WARNING: file ${conf_file} not found!"
-			fi
-		fi
-	fi
-}
-
 # CPU hotplug helper function that turns on/off the CPU cores in the specified range
 # * Parameters:
 # * ${1}: desired online status (0 for offline, 1 for online)
@@ -680,8 +528,6 @@ SOCFAMILY=""
 
 set_socfamily
 set_power_state_perm
-create_nvpmodel_symlink
-create_nvfancontrol_symlink
 cpu_hotplug
 set_cpu_governor
 set_cpu_floor_freq
-- 
2.47.1

