Index: edk2-tegra/edk2-nvidia/Silicon/NVIDIA/Library/UsbFalconLib/UsbFalconLib.c
===================================================================
--- edk2-tegra.orig/edk2-nvidia/Silicon/NVIDIA/Library/UsbFalconLib/UsbFalconLib.c
+++ edk2-tegra/edk2-nvidia/Silicon/NVIDIA/Library/UsbFalconLib/UsbFalconLib.c
@@ -9,12 +9,12 @@
 **/
 
 #include <PiDxe.h>
+#include <Library/BaseMemoryLib.h>
 #include <Library/BaseLib.h>
 #include <Library/DebugLib.h>
 #include <Library/IoLib.h>
 #include <Library/UsbFalconLib.h>
 #include <Library/UefiBootServicesTableLib.h>
-#include <string.h>
 #include <Library/DmaLib.h>
 
 
@@ -221,8 +221,8 @@ FalconFirmwareIfrLoad (
   }
 
   DEBUG ((EFI_D_ERROR, "%a: Firmware %p FirmwareSize %x (unaligned)\r\n",__FUNCTION__, Firmware, FirmwareSize));
-  memset (FirmwareBuffer, 0xdf, BufferSize);
-  memcpy (FirmwareBuffer, Firmware, FirmwareSize);
+  SetMem (FirmwareBuffer, BufferSize, 0xdf);
+  CopyMem (FirmwareBuffer, Firmware, FirmwareSize);
   for (i = 0; i < FirmwareSize; i++)
   {
     if (FirmwareBuffer[i] != Firmware[i])
@@ -315,8 +315,8 @@ FalconFirmwareLoad (
   }
 
   DEBUG ((EFI_D_VERBOSE, "%a: Firmware %p FirmwareSize %x (unaligned)\r\n",__FUNCTION__, Firmware, FirmwareSize));
-  memset (FirmwareBuffer, 0xdf, BufferSize);
-  memcpy (FirmwareBuffer, Firmware, FirmwareSize);
+  SetMem (FirmwareBuffer, BufferSize, 0xdf);
+  CopyMem (FirmwareBuffer, Firmware, FirmwareSize);
   for (i = 0; i < FirmwareSize; i++)
   {
     if (FirmwareBuffer[i] != Firmware[i])
