From 885fe8d6600f9ed8075fac1c86065da578d69e3e Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Thu, 15 Jan 2026 12:45:54 -0800
Subject: [PATCH] Makefile: Add OOTSRC for nvdisplay builds

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Matt Madison <matt@madison.systems>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index b915c67..9979b43 100644
--- a/Makefile
+++ b/Makefile
@@ -134,6 +134,7 @@ define display-cmd
 		SYSSRCNVOOT=$(MAKEFILE_DIR)/nvidia-oot \
 		SYSSRC=$(KERNEL_SRC) \
 		SYSOUT=$(KBUILD_OUTPUT) \
+		OOTSRC=$(MAKEFILE_DIR)/nvidia-oot \
 		KBUILD_EXTRA_SYMBOLS=$(MAKEFILE_DIR)/nvidia-oot/Module.symvers \
 		CC="$(CC)" \
 		LD="$(LD)" \
--
2.34.1

