From aff20e7425f18828efd4a1bc3f7b53da08d77025 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Mon, 11 Aug 2025 16:47:59 +0100
Subject: [PATCH 5/6] Fix unifiedgpudisp conftest gcc 14 compatibility issues

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 unifiedgpudisp/kernel-open/conftest.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/unifiedgpudisp/kernel-open/conftest.sh b/unifiedgpudisp/kernel-open/conftest.sh
index 3f42ff2..12acf19 100755
--- a/unifiedgpudisp/kernel-open/conftest.sh
+++ b/unifiedgpudisp/kernel-open/conftest.sh
@@ -106,8 +106,8 @@ build_cflags() {
     ISYSTEM=`$CC -print-file-name=include 2> /dev/null`
     BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
--nostdinc -isystem $ISYSTEM \
--Wno-implicit-function-declaration -Wno-strict-prototypes"
+-nostdinc -isystem $ISYSTEM -fshort-wchar \
+-Wno-implicit-function-declaration -Wno-strict-prototypes -Wno-error=incompatible-pointer-types"
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
         OUTPUT_CFLAGS="-I$OUTPUT/include2 -I$OUTPUT/include"
-- 
2.34.1

