From 714200ba16cba8ceef60fd0c7dcb8ab72112c6e3 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Mon, 11 Aug 2025 16:47:59 +0100
Subject: [PATCH 09/11] Fix unifiedgpudisp conftest gcc 14 compatibility issues

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 kernel-open/conftest.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel-open/conftest.sh b/kernel-open/conftest.sh
index 9c34e18..e369be8 100755
--- a/kernel-open/conftest.sh
+++ b/kernel-open/conftest.sh
@@ -106,8 +106,8 @@ build_cflags() {
     ISYSTEM=`$CC -print-file-name=include 2> /dev/null`
     BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
--nostdinc -isystem $ISYSTEM \
--Wno-implicit-function-declaration -Wno-strict-prototypes"
+-nostdinc -isystem $ISYSTEM -fshort-wchar \
+-Wno-implicit-function-declaration -Wno-strict-prototypes -Wno-error=incompatible-pointer-types"
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
         OUTPUT_CFLAGS="-I$OUTPUT/include2 -I$OUTPUT/include"
-- 
2.43.0

