From 441423420991d8afced04b784ad21ef6170b744b Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Mon, 14 Oct 2024 02:19:11 -0700
Subject: [PATCH] Fixups for cross building in OE

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 python/CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 6666511d..4b003d20 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -319,11 +319,6 @@ set(CPP_STANDARD
     CACHE STRING "CPP Standard Version")
 set(CMAKE_CXX_STANDARD ${CPP_STANDARD})

-if(NOT MSVC)
-    # This allows us to use TRT libs shipped with standalone wheels.
-    set(CMAKE_SHARED_LINKER_FLAGS -Wl,-rpath=$ORIGIN:$ORIGIN/../${TENSORRT_MODULE}_libs)
-endif()
-
 # -------- PATHS --------
 message(STATUS "EXT_PATH: ${EXT_PATH}")
 message(STATUS "TENSORRT_BUILD: ${TENSORRT_BUILD}")
--