Index: gst-egl/ext/eglgles/gsteglglessink.c
===================================================================
--- gst-egl.orig/ext/eglgles/gsteglglessink.c
+++ gst-egl/ext/eglgles/gsteglglessink.c
@@ -3318,7 +3318,7 @@ eglglessink_plugin_init (GstPlugin * plu
 /* gstreamer looks for this structure to register eglglessinks */
 GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
     GST_VERSION_MINOR,
-    eglglessink,
+    nveglglessink,
     "EGL/GLES sink",
     eglglessink_plugin_init,
     VERSION, GST_LICENSE, GST_PACKAGE_NAME, GST_PACKAGE_ORIGIN)
