From 4deb28939dbdc025c87fd316a89068d1af2592c5 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Fri, 27 Mar 2026 14:41:55 -0700
Subject: [PATCH] Fix builds without x11

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Kurt Kiefer <kurt.kiefer@arthrex.com>
Signed-off-by: Matt Madison <matt@madison.systems>
---
 gst-egl/ext/eglgles/gsteglglessink.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gst-egl/ext/eglgles/gsteglglessink.c b/gst-egl/ext/eglgles/gsteglglessink.c
index 1ad4ddb..c70ee37 100644
--- a/gst-egl/ext/eglgles/gsteglglessink.c
+++ b/gst-egl/ext/eglgles/gsteglglessink.c
@@ -107,7 +107,9 @@
 #include <gst/video/gstvideopool.h>
 #include <gst/video/videooverlay.h>
 
+#ifdef USE_EGL_X11
 #include <X11/Xlib.h>
+#endif
 
 #include "gstegladaptation.h"
 #include "video_platform_wrapper.h"
-- 
2.43.0

