From 51d489751f048e727509bc121c9e09b5b57e0ee2 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Fri, 27 Mar 2026 14:42:22 -0700
Subject: [PATCH] Fix builds without wayland

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
Signed-off-by: Matt Madison <matt@madison.systems>
---
 gst-egl/ext/eglgles/gstegladaptation_egl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gst-egl/ext/eglgles/gstegladaptation_egl.c b/gst-egl/ext/eglgles/gstegladaptation_egl.c
index 8a4abef..77d0501 100644
--- a/gst-egl/ext/eglgles/gstegladaptation_egl.c
+++ b/gst-egl/ext/eglgles/gstegladaptation_egl.c
@@ -283,9 +283,11 @@ gboolean
 gst_egl_adaptation_context_swap_buffers (GstEglAdaptationContext * ctx, gchar* winsys,
         gpointer * own_window_data, GstBuffer * buf, gboolean show_latency)
 {
+#ifdef USE_EGL_WAYLAND
   if (g_strcmp0(winsys, "wayland") == 0 && show_latency) {
     register_presentation_feedback(own_window_data, buf);
   }
+#endif
 
   gboolean ret = eglSwapBuffers (gst_egl_display_get (ctx->display),
       ctx->eglglesctx->surface);
-- 
2.43.0

