From 7ab0e7ceafb61b825a58f112932a0476cd87b857 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Tue, 17 Nov 2020 06:23:04 -0800
Subject: [PATCH 7/9] t186/l4t.mk: make some build options configurable

Signed-off-by: Matt Madison <matt@madison.systems>
---
 bootloader/partner/t18x/cboot/platform/t186/l4t.mk | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/bootloader/partner/t18x/cboot/platform/t186/l4t.mk b/bootloader/partner/t18x/cboot/platform/t186/l4t.mk
index 9792b32..d07b2cb 100644
--- a/bootloader/partner/t18x/cboot/platform/t186/l4t.mk
+++ b/bootloader/partner/t18x/cboot/platform/t186/l4t.mk
@@ -11,15 +11,19 @@
 MODULE_DEPS += \
 	$(LOCAL_DIR)/../../../../common/drivers/crypto
 
+EXTRA_GLOBAL_DEFINES ?= \
+	CONFIG_ENABLE_DISPLAY=1 \
+	CONFIG_ENABLE_L4T_RECOVERY=1
+
 # Add any needed GLOBAL_DEFINES here
 GLOBAL_DEFINES += \
 	CONFIG_OS_IS_L4T=1 \
 	CONFIG_ENABLE_SATA=1 \
 	CONFIG_ENABLE_SDCARD=1 \
-	CONFIG_ENABLE_DISPLAY=1 \
 	CONFIG_ENABLE_SECURE_BOOT=1 \
 	CONFIG_INITIALIZE_DISPLAY=1 \
 	CONFIG_DYNAMIC_LOAD_ADDRESS=1 \
-	CONFIG_ENABLE_L4T_RECOVERY=1 \
-	CONFIG_ENABLE_DP=1
+	CONFIG_ENABLE_DP=1 \
+	$(EXTRA_GLOBAL_DEFINES)
+
 # 0-DSI, 1-HDMI, 2-DP, 3-EDP
-- 
2.27.0

