From 843ca0b598d1346f3eb87aa624821051afaaafba Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Wed, 28 Feb 2024 23:08:26 +0000
Subject: [PATCH 1/2] Patch udev rules for OE use

Upstream-Status: Inappropriate [OE-Specific]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 etc/udev/rules.d/99-tegra-devices.rules | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/etc/udev/rules.d/99-tegra-devices.rules b/etc/udev/rules.d/99-tegra-devices.rules
index a655804..2b16bfd 100644
--- a/etc/udev/rules.d/99-tegra-devices.rules
+++ b/etc/udev/rules.d/99-tegra-devices.rules
@@ -9,7 +9,6 @@
 # its affiliates is strictly prohibited.
 
 ACTION=="remove" GOTO="nvidia_end"
-KERNEL=="camera.pcl", RUN+="/usr/sbin/camera_device_detect"
 
 # root only devices
 KERNEL=="knvrm" OWNER="root" GROUP="root" MODE="0660"
@@ -73,8 +72,4 @@ KERNEL=="camchar*" OWNER="root" GROUP="video" MODE="0660"
 KERNEL=="capture-*" OWNER="root" GROUP="video" MODE="0660"
 KERNEL=="cdi_tsc" OWNER="root" GROUP="video" MODE="0660"
 
-# security related devices
-KERNEL=="tegra-crypto" OWNER="root" GROUP="crypto" MODE="0660"
-KERNEL=="trusty-ipc-dev*" OWNER="root" GROUP="trusty" MODE="0660"
-
 LABEL="nvidia_end"
-- 
2.34.1

