From c2e973957f9fcbedd0bcf16bcc3af9aa51729f2b Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Tue, 16 Nov 2021 09:08:15 -0800
Subject: [PATCH] Patch udev rules for OE use

Signed-off-by: Matt Madison <matt@madison.systems>
---
 etc/udev/rules.d/99-tegra-devices.rules | 5 -----
 1 file changed, 5 deletions(-)

Index: tegra-configs-35.4.1-20230801124926/etc/udev/rules.d/99-tegra-devices.rules
===================================================================
--- tegra-configs-35.4.1-20230801124926.orig/etc/udev/rules.d/99-tegra-devices.rules
+++ tegra-configs-35.4.1-20230801124926/etc/udev/rules.d/99-tegra-devices.rules
@@ -3,7 +3,6 @@
 #
 
 ACTION=="remove" GOTO="nvidia_end"
-KERNEL=="camera.pcl", RUN+="/usr/sbin/camera_device_detect"
 
 # root only devices
 KERNEL=="knvrm" OWNER="root" GROUP="root" MODE="0660"
@@ -67,8 +66,4 @@ KERNEL=="camchar*" OWNER="root" GROUP="v
 KERNEL=="capture-*" OWNER="root" GROUP="video" MODE="0660"
 KERNEL=="cdi_tsc" OWNER="root" GROUP="video" MODE="0660"
 
-# security related devices
-KERNEL=="tegra-crypto" OWNER="root" GROUP="crypto" MODE="0660"
-KERNEL=="trusty-ipc-dev*" OWNER="root" GROUP="trusty" MODE="0660"
-
 LABEL="nvidia_end"
