From a6483bf9114ca761a0ffeedeedb387ca0531a74d Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Tue, 16 Nov 2021 09:08:15 -0800
Subject: [PATCH] Patch udev rules for OE use

Signed-off-by: Matt Madison <matt@madison.systems>

---
 etc/udev/rules.d/99-tegra-devices.rules | 5 -----
 1 file changed, 5 deletions(-)

Index: tegra-configs-36.3.0-20240424200557/etc/udev/rules.d/99-tegra-devices.rules
===================================================================
--- tegra-configs-36.3.0-20240424200557.orig/etc/udev/rules.d/99-tegra-devices.rules
+++ tegra-configs-36.3.0-20240424200557/etc/udev/rules.d/99-tegra-devices.rules
@@ -9,7 +9,6 @@
 # its affiliates is strictly prohibited.
 
 ACTION=="remove" GOTO="nvidia_end"
-KERNEL=="camera.pcl", RUN+="/usr/sbin/camera_device_detect"
 
 # root only devices
 KERNEL=="knvrm" OWNER="root" GROUP="root" MODE="0660"
