From a5634a1e294bf8bd76542af5848c2750805906d5 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Sun, 10 Dec 2023 18:04:47 +0000
Subject: [PATCH] Patch nv.sh script for OE use

Signed-off-by: Matt Madison <matt@madison.systems>
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 etc/systemd/nv.sh | 219 ----------------------------------------------
 1 file changed, 219 deletions(-)

diff --git a/etc/systemd/nv.sh b/etc/systemd/nv.sh
index 8044f43..007ddc3 100755
--- a/etc/systemd/nv.sh
+++ b/etc/systemd/nv.sh
@@ -92,10 +92,6 @@ if [ -e "/proc/device-tree/compatible" ]; then
 	fi
 fi
 
-if [ "$SOCFAMILY" = "tegra194" ]; then
-	sudo bash -c "ln -sf /etc/X11/xorg.conf.t194_ref /etc/X11/xorg.conf"
-fi
-
 if [ "$SOCFAMILY" = "Tegra13" ] &&
 	[ -e /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq ]; then
 	bash \
@@ -115,11 +111,6 @@ elif [ -w /sys/module/cpu_tegra3/parameters/auto_hotplug ]; then
 	echo 1 > /sys/module/cpu_tegra3/parameters/auto_hotplug
 fi
 
-# Remove the spawning of ondemand service
-if [ -e "/etc/systemd/system/multi-user.target.wants/ondemand.service" ]; then
-	rm -f "/etc/systemd/system/multi-user.target.wants/ondemand.service"
-fi
-
 # mmc read ahead size
 if [ -e /sys/block/mmcblk0/queue/read_ahead_kb ]; then
 	echo 2048 > /sys/block/mmcblk0/queue/read_ahead_kb
@@ -135,13 +126,6 @@ if [ "${SOCFAMILY}" = "tegra194" ]; then
 	fi
 fi
 
-# Ensure libglx.so is not overwritten by a distribution update of Xorg
-# Alternatively, package management tools could be used to prevent updates
-ARCH=$(/usr/bin/dpkg --print-architecture)
-if [ "${ARCH}" = "arm64" ]; then
-	LIB_DIR="/usr/lib/aarch64-linux-gnu"
-fi
-
 # Disable lazy vfree pages
 if [ -e "/proc/sys/vm/lazy_vfree_pages" ]; then
 	echo 0 > "/proc/sys/vm/lazy_vfree_pages"
@@ -189,206 +173,3 @@ fi
 if [ -e "/var/lib/lightdm" ]; then
 	sudo chown lightdm:lightdm /var/lib/lightdm -R
 fi
-
-# Set INSTALL_DOWNSTREAM_WESTON="1" to install downstream weston. This will
-# overwrite weston binaries in standard weston installation path with downstream
-# weston binaries which are present in "/usr/lib/aarch64-linux-gnu/nvidia. This is default.
-#
-# Set INSTALL_DOWNSTREAM_WESTON="0" to avoid overwriting weston binaries in standard
-# weston installation path with downstream weston binaries which are present in
-# /usr/lib/aarch64-linux-gnu/nvidia.
-INSTALL_DOWNSTREAM_WESTON="1"
-if  [ "${INSTALL_DOWNSTREAM_WESTON}" -eq "1" ]; then
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/desktop-shell.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/desktop-shell.so" "${LIB_DIR}/weston/desktop-shell.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/drm-backend.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/drm-backend.so" "${LIB_DIR}/weston/drm-backend.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/fullscreen-shell.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/fullscreen-shell.so" "${LIB_DIR}/weston/fullscreen-shell.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/gl-renderer.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/gl-renderer.so" "${LIB_DIR}/weston/gl-renderer.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/hmi-controller.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/hmi-controller.so" "${LIB_DIR}/weston/hmi-controller.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/ivi-controller.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/ivi-controller.so" "${LIB_DIR}/weston/ivi-controller.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/ivi-shell.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/ivi-shell.so" "${LIB_DIR}/weston/ivi-shell.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/libweston-6.so.0" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/libweston-6.so.0" "${LIB_DIR}/libweston-6.so.0"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/libweston-desktop-6.so.0" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/libweston-desktop-6.so.0" "${LIB_DIR}/libweston-desktop-6.so.0"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/wayland-backend.so" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/wayland-backend.so" "${LIB_DIR}/weston/wayland-backend.so"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmClient.so.2.2.0" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmClient.so.2.2.0" "${LIB_DIR}/libilmClient.so.2.2.0"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmCommon.so.2.2.0" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmCommon.so.2.2.0" "${LIB_DIR}/libilmCommon.so.2.2.0"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmControl.so.2.2.0" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmControl.so.2.2.0" "${LIB_DIR}/libilmControl.so.2.2.0"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmInput.so.2.2.0" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/libilmInput.so.2.2.0" "${LIB_DIR}/libilmInput.so.2.2.0"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-desktop-shell" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-desktop-shell" "/usr/lib/weston/weston-desktop-shell"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-keyboard" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-keyboard" "/usr/lib/weston/weston-keyboard"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-screenshooter" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-screenshooter" "/usr/lib/westonweston-screenshooter"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/EGLWLInputEventExample" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/EGLWLInputEventExample" "/usr/bin/EGLWLInputEventExample"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/EGLWLMockNavigation" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/EGLWLMockNavigation" "/usr/bin/EGLWLMockNavigation"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/LayerManagerControl" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/LayerManagerControl" "/usr/bin/LayerManagerControl"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/simple-weston-client" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/simple-weston-client" "/usr/bin/simple-weston-client"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/spring-tool" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/spring-tool" "/usr/bin/spring-tool"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston" "/usr/bin/weston"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-calibrator" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-calibrator" "/usr/bin/weston-calibrator"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-clickdot" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-clickdot" "/usr/bin/weston-clickdot"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-cliptest" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-cliptest" "/usr/bin/weston-cliptest"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-content-protection" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-content-protection" "/usr/bin/weston-content-protection"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-debug" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-debug" "/usr/bin/weston-debug"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-dnd" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-dnd" "/usr/bin/weston-dnd"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-eventdemo" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-eventdemo" "/usr/bin/weston-eventdemo"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-flower" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-flower" "/usr/bin/weston-flower"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-fullscreen" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-fullscreen" "/usr/bin/weston-fullscreen"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-image" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-image" "/usr/bin/weston-image"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-info" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-info" "/usr/bin/weston-info"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-ivi-shell-user-interface" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-ivi-shell-user-interface" "/usr/bin/weston-ivi-shell-user-interface"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-launch" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-launch" "/usr/bin/weston-launch"
-		chmod "+s" "/usr/bin/weston-launch"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-multi-resource" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-multi-resource" "/usr/bin/weston-multi-resource"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-output-mode" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-output-mode" "/usr/bin/weston-output-mode"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-resizor" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-resizor" "/usr/bin/weston-resizor"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-scaler" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-scaler" "/usr/bin/weston-scaler"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-simple-egl" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-simple-egl" "/usr/bin/weston-simple-egl"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-simple-shm" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-simple-shm" "/usr/bin/weston-simple-shm"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-simple-touch" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-simple-touch" "/usr/bin/weston-simple-touch"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-smoke" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-smoke" "/usr/bin/weston-smoke"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-stacking" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-stacking" "/usr/bin/weston-stacking"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-subsurfaces" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-subsurfaces" "/usr/bin/weston-subsurfaces"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-terminal" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-terminal" "/usr/bin/weston-terminal"
-	fi
-
-	if [ -e "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-transformed" ]; then
-		ln -sf "/usr/lib/aarch64-linux-gnu/nvidia/weston/weston-transformed" "/usr/bin/weston-transformed"
-	fi
-fi
-- 
2.34.1

