From b1a650c065b101083dcac9e8d885c26b0d15686f Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Wed, 28 Feb 2024 23:12:36 +0000
Subject: [PATCH 2/2] Patch nv.sh script for OE use

Upstream-Status: Inappropriate [OE-Specific]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 etc/systemd/nv.sh | 264 ----------------------------------------------
 1 file changed, 264 deletions(-)

diff --git a/etc/systemd/nv.sh b/etc/systemd/nv.sh
index 6fddc2c..a8da38f 100755
--- a/etc/systemd/nv.sh
+++ b/etc/systemd/nv.sh
@@ -105,12 +105,6 @@ if [ -e "/proc/device-tree/compatible" ]; then
 	fi
 fi
 
-if [ "$SOCFAMILY" = "tegra194" ]; then
-	sudo bash -c "ln -sf /etc/X11/xorg.conf.t194_ref /etc/X11/xorg.conf"
-elif [ "$machine" = "jetson_tx1" ] || [ "$machine" = "jetson_e" ]; then
-	sudo bash -c "ln -sf /etc/X11/xorg.conf.jetson_e /etc/X11/xorg.conf"
-fi
-
 if [ "$SOCFAMILY" = "Tegra13" ] &&
 	[ -e /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq ]; then
 	bash \
@@ -133,11 +127,6 @@ if [ "$SOCFAMILY" != "tegra210" ]; then
 	fi
 fi
 
-# Remove the spawning of ondemand service
-if [ -e "/etc/systemd/system/multi-user.target.wants/ondemand.service" ]; then
-	rm -f "/etc/systemd/system/multi-user.target.wants/ondemand.service"
-fi
-
 # mmc read ahead size
 if [ -e /sys/block/mmcblk0/queue/read_ahead_kb ]; then
 	echo 2048 > /sys/block/mmcblk0/queue/read_ahead_kb
@@ -153,260 +142,7 @@ if [ "$SOCFAMILY" = "tegra186" ] || [ "$SOCFAMILY" = "tegra210" ] || [ "$SOCFAMI
 	fi
 fi
 
-# Ensure libglx.so is not overwritten by a distribution update of Xorg
-# Alternatively, package management tools could be used to prevent updates
-ARCH=$(/usr/bin/dpkg --print-architecture)
-if [ "x${ARCH}" = "xarm64" ]; then
-	LIB_DIR="/usr/lib/aarch64-linux-gnu"
-fi
-
 # Disable lazy vfree pages
 if [ -e "/proc/sys/vm/lazy_vfree_pages" ]; then
 	echo 0 > "/proc/sys/vm/lazy_vfree_pages"
 fi
-
-# WAR for https://bugs.launchpad.net/ubuntu/+source/mesa/+bug/1776499
-# and https://bugs.launchpad.net/ubuntu/+source/mesa/+bug/1946621
-# These issues are fixed from MESA version 21.2 and so detect the MESA
-# version and apply the workaround if needed.
-# When DISABLE_MESA_EGL="1" glvnd will not load mesa EGL library.
-# When DISABLE_MESA_EGL="0" glvnd will load mesa EGL library.
-# nvidia EGL library is prioritized over mesa even if DISABLE_MESA_EGL="0".
-MESA_VERSION="$(dpkg -s libegl-mesa0 | sed -nr 's/^Version: ([0-9]+\.[0-9]+)\.[0-9]+-.*+ubuntu.*/\1/p')"
-DISABLE_MESA_EGL="0"
-
-if [ "${MESA_VERSION}" != "" ]; then
-    DISABLE_MESA_EGL="$(echo "${MESA_VERSION} < 21.2" | bc)"
-fi
-
-if [ -f "/usr/share/glvnd/egl_vendor.d/50_mesa.json" ]; then
-	if  [ "${DISABLE_MESA_EGL}" -eq "1" ]; then
-		sed -i "s/\"library_path\" : .*/\"library_path\" : \"\"/g" \
-			"/usr/share/glvnd/egl_vendor.d/50_mesa.json"
-	else
-		sed -i "s/\"library_path\" : .*/\"library_path\" : \"libEGL_mesa.so.0\"/g" \
-			"/usr/share/glvnd/egl_vendor.d/50_mesa.json"
-	fi
-fi
-
-# Add gdm in video group
-getent passwd gdm > /dev/null
-if [ $? -eq 0 ]; then
-	adduser gdm video
-fi
-
-# Add lightdm in video group
-grep "lightdm" "/etc/group" > /dev/null
-if [ $? -eq 0 ]; then
-	groups "lightdm" | grep "video" > /dev/null
-	if [ $? -eq 1 ]; then
-		addgroup "lightdm" "video"
-	fi
-fi
-
-if [ -e "/var/lib/lightdm" ]; then
-	sudo chown lightdm:lightdm /var/lib/lightdm -R
-fi
-
-# Set INSTALL_DOWNSTREAM_WESTON="1" to install downstream weston. This will
-# overwrite weston binaries in standard weston installation path with downstream
-# weston binaries which are present in ${LIB_DIR}/tegra. This is default.
-#
-# Set INSTALL_DOWNSTREAM_WESTON="0" to avoid overwriting weston binaries in standard
-# weston installation path with downstream weston binaries which are present in
-# ${LIB_DIR}/tegra.
-INSTALL_DOWNSTREAM_WESTON="1"
-if  [ "${INSTALL_DOWNSTREAM_WESTON}" -eq "1" ]; then
-	if [ -e "${LIB_DIR}/tegra/weston/desktop-shell.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/desktop-shell.so" "${LIB_DIR}/weston/desktop-shell.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/drm-backend.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/drm-backend.so" "${LIB_DIR}/weston/drm-backend.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/fullscreen-shell.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/fullscreen-shell.so" "${LIB_DIR}/weston/fullscreen-shell.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/gl-renderer.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/gl-renderer.so" "${LIB_DIR}/weston/gl-renderer.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/hmi-controller.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/hmi-controller.so" "${LIB_DIR}/weston/hmi-controller.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/ivi-controller.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/ivi-controller.so" "${LIB_DIR}/weston/ivi-controller.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/ivi-shell.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/ivi-shell.so" "${LIB_DIR}/weston/ivi-shell.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/libweston-6.so.0" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/libweston-6.so.0" "${LIB_DIR}/libweston-6.so.0"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/libweston-desktop-6.so.0" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/libweston-desktop-6.so.0" "${LIB_DIR}/libweston-desktop-6.so.0"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/wayland-backend.so" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/wayland-backend.so" "${LIB_DIR}/weston/wayland-backend.so"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/libilmClient.so.2.2.0" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/libilmClient.so.2.2.0" "${LIB_DIR}/libilmClient.so.2.2.0"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/libilmCommon.so.2.2.0" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/libilmCommon.so.2.2.0" "${LIB_DIR}/libilmCommon.so.2.2.0"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/libilmControl.so.2.2.0" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/libilmControl.so.2.2.0" "${LIB_DIR}/libilmControl.so.2.2.0"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/libilmInput.so.2.2.0" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/libilmInput.so.2.2.0" "${LIB_DIR}/libilmInput.so.2.2.0"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-desktop-shell" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-desktop-shell" "/usr/lib/weston/weston-desktop-shell"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-keyboard" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-keyboard" "/usr/lib/weston/weston-keyboard"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-screenshooter" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-screenshooter" "/usr/lib/westonweston-screenshooter"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/EGLWLInputEventExample" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/EGLWLInputEventExample" "/usr/bin/EGLWLInputEventExample"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/EGLWLMockNavigation" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/EGLWLMockNavigation" "/usr/bin/EGLWLMockNavigation"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/LayerManagerControl" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/LayerManagerControl" "/usr/bin/LayerManagerControl"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/simple-weston-client" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/simple-weston-client" "/usr/bin/simple-weston-client"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/spring-tool" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/spring-tool" "/usr/bin/spring-tool"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston" "/usr/bin/weston"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-calibrator" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-calibrator" "/usr/bin/weston-calibrator"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-clickdot" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-clickdot" "/usr/bin/weston-clickdot"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-cliptest" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-cliptest" "/usr/bin/weston-cliptest"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-content-protection" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-content-protection" "/usr/bin/weston-content-protection"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-debug" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-debug" "/usr/bin/weston-debug"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-dnd" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-dnd" "/usr/bin/weston-dnd"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-eventdemo" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-eventdemo" "/usr/bin/weston-eventdemo"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-flower" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-flower" "/usr/bin/weston-flower"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-fullscreen" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-fullscreen" "/usr/bin/weston-fullscreen"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-image" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-image" "/usr/bin/weston-image"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-info" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-info" "/usr/bin/weston-info"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-ivi-shell-user-interface" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-ivi-shell-user-interface" "/usr/bin/weston-ivi-shell-user-interface"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-launch" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-launch" "/usr/bin/weston-launch"
-		chmod "+s" "/usr/bin/weston-launch"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-multi-resource" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-multi-resource" "/usr/bin/weston-multi-resource"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-output-mode" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-output-mode" "/usr/bin/weston-output-mode"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-resizor" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-resizor" "/usr/bin/weston-resizor"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-scaler" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-scaler" "/usr/bin/weston-scaler"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-simple-egl" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-simple-egl" "/usr/bin/weston-simple-egl"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-simple-shm" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-simple-shm" "/usr/bin/weston-simple-shm"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-simple-touch" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-simple-touch" "/usr/bin/weston-simple-touch"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-smoke" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-smoke" "/usr/bin/weston-smoke"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-stacking" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-stacking" "/usr/bin/weston-stacking"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-subsurfaces" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-subsurfaces" "/usr/bin/weston-subsurfaces"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-terminal" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-terminal" "/usr/bin/weston-terminal"
-	fi
-
-	if [ -e "${LIB_DIR}/tegra/weston/weston-transformed" ]; then
-		ln -sf "${LIB_DIR}/tegra/weston/weston-transformed" "/usr/bin/weston-transformed"
-	fi
-fi
-- 
2.34.1

