From 5ea4e93464f0c51b1e321a90c2c1f4d8f22747f6 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Thu, 16 Jul 2020 06:28:39 -0700
Subject: [PATCH] Build fixups

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>

---
 Makefile | 23 ++++++++++-------------
 1 file changed, 10 insertions(+), 13 deletions(-)

Index: gst-nvv4l2camera/Makefile
===================================================================
--- gst-nvv4l2camera.orig/Makefile
+++ gst-nvv4l2camera/Makefile
@@ -27,9 +27,10 @@
 
 SO_NAME := libgstnvv4l2camerasrc.so
 
-CC := g++
+exec_prefix ?= /usr
+libdir ?= $(exec_prefix)/lib
 
-TARGET_DEVICE = $(shell gcc -dumpmachine | cut -f1 -d -)
+GST_INSTALL_DIR ?= $(libdir)/gstreamer-1.0
 
 ifdef CUDA_VER
 CUDA_PATH := /usr/local/cuda-$(CUDA_VER)
@@ -37,31 +38,13 @@ else
 CUDA_PATH := /usr/local/cuda
 endif
 
-# Location of the target rootfs
-ifeq ($(shell uname -m), aarch64)
-TARGET_ROOTFS :=
-else
-ifeq ($(TARGET_ROOTFS),)
-$(error Please specify the target rootfs path if you are cross-compiling)
-endif
-endif
-
 ALGO_CUDA_DIR=cuda
 
-GST_INSTALL_DIR?=/usr/lib/aarch64-linux-gnu/gstreamer-1.0/
-LIB_INSTALL_DIR?=/usr/lib/aarch64-linux-gnu/tegra/
-CFLAGS:=
-LIBS:= -lnvbufsurface -lnvbufsurftransform \
-	-lcuda -lcudart \
-	-L"$(TARGET_ROOTFS)/$(CUDA_PATH)/lib64" \
-
+LIBS:= -lnvbufsurface -lnvbufsurftransform
 
 SRCS := $(wildcard *.cpp)
 
-INCLUDES += -I./ -I../ -I./cuda \
-	-I/usr/src/jetson_multimedia_api/include/ \
-	-I$(CUDA_PATH)/include
-
+INCLUDES += -I./cuda
 
 PKGS := gstreamer-1.0 \
 	gstreamer-base-1.0 \
@@ -69,37 +52,45 @@ PKGS := gstreamer-1.0 \
 	gstreamer-allocators-1.0 \
 	glib-2.0 \
 	libv4l2
+ifdef CUDA_VER
+PKGS += cuda-$(CUDA_VER) cudart-$(CUDA_VER)
+else
+CUDA_PATH := /usr/loca/cuda
+INCLUDES += -I$(CUDA_PATH)/include
+LIBS += -lcuda -lcudart
+endif
 
 OBJS := $(SRCS:.cpp=.o)
 OBJS += $(ALGO_CUDA_DIR)/NvCudaProc.o
 
 
-CFLAGS += -fPIC \
+CXXFLAGS += -fPIC \
 	-DEXPLICITLY_ADDED=1 \
         -DGETTEXT_PACKAGE=1 \
         -DHAVE_LIBV4L2=1 \
         -DUSE_V4L2_TARGET_NV=1
 
-CFLAGS += `pkg-config --cflags $(PKGS)`
+CXXFLAGS += `pkg-config --cflags $(PKGS)`
 
-LDFLAGS = -Wl,--no-undefined -L$(LIB_INSTALL_DIR) -Wl,-rpath,$(LIB_INSTALL_DIR)
+LDFLAGS += -Wl,--no-undefined
 
 LIBS += `pkg-config --libs $(PKGS)`
 
 all: $(SO_NAME)
 
 %.o: %.cpp
-	$(CC) -c $< $(CFLAGS) $(INCLUDES) -o $@
+	$(CXX) $(CXXFLAGS) $(INCLUDES) -c $<  -o $@
 
 $(ALGO_CUDA_DIR)/%.o: $(ALGO_CUDA_DIR)/%.cpp
-	$(CC) -c $< $(CFLAGS) $(INCLUDES) -o $@
+	$(CXX) $(CXXFLAGS) $(INCLUDES) -c $< -o $@
 
 $(SO_NAME): $(OBJS)
-	$(CC) -shared -o $(SO_NAME) $(OBJS) $(LIBS) $(LDFLAGS)
+	$(CXX) -shared -o $(SO_NAME) $(OBJS) $(LIBS) $(LDFLAGS)
 
 .PHONY: install
 install: $(SO_NAME)
-	cp -vp $(SO_NAME) $(GST_INSTALL_DIR)
+	install -d $(DESTDIR)$(GST_INSTALL_DIR)
+	install -m 0644 $(SO_NAME) $(DESTDIR)$(GST_INSTALL_DIR)/
 
 .PHONY: clean
 clean:
