From e0ebc42e239fb58ea2eb17fa646782af2c6db8db Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Thu, 16 Jul 2020 10:43:09 -0700
Subject: [PATCH] Clean up compiler warnings

from a compilation pass with -Wall -Wpedantic.

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>

---
 gstnvv4l2camerasrc.cpp | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

Index: gst-nvv4l2camera/gstnvv4l2camerasrc.cpp
===================================================================
--- gst-nvv4l2camera.orig/gstnvv4l2camerasrc.cpp
+++ gst-nvv4l2camera/gstnvv4l2camerasrc.cpp
@@ -119,11 +119,11 @@ GType gst_nv_memory_allocator_get_type (
 #define GST_TYPE_NV_MEMORY_ALLOCATOR   (gst_nv_memory_allocator_get_type())
 
 #define gst_nvv4l2camera_buffer_pool_parent_class bpool_parent_class
-G_DEFINE_TYPE_WITH_CODE (GstNvV4l2CameraBufferPool, gst_nvv4l2camera_buffer_pool, GST_TYPE_BUFFER_POOL, G_ADD_PRIVATE(GstNvV4l2CameraBufferPool));
+G_DEFINE_TYPE_WITH_CODE (GstNvV4l2CameraBufferPool, gst_nvv4l2camera_buffer_pool, GST_TYPE_BUFFER_POOL, G_ADD_PRIVATE(GstNvV4l2CameraBufferPool))
 
 #define gst_nv_v4l2_camera_src_parent_class parent_class
-G_DEFINE_TYPE (GstNvV4l2CameraSrc, gst_nv_v4l2_camera_src, GST_TYPE_BASE_SRC);
-G_DEFINE_TYPE (GstNVV4l2MemoryAllocator, gst_nv_memory_allocator, GST_TYPE_ALLOCATOR);
+G_DEFINE_TYPE (GstNvV4l2CameraSrc, gst_nv_v4l2_camera_src, GST_TYPE_BASE_SRC)
+G_DEFINE_TYPE (GstNVV4l2MemoryAllocator, gst_nv_memory_allocator, GST_TYPE_ALLOCATOR)
 
 #define GST_NVMEMORY_ALLOCATOR(obj) (G_TYPE_CHECK_INSTANCE_CAST((obj),GST_TYPE_NV_MEMORY_ALLOCATOR,GstNVV4l2MemoryAllocator))
 
@@ -691,7 +691,7 @@ wrong_video_caps:
   {
     GST_OBJECT_UNLOCK (pool);
     GST_WARNING_OBJECT (pool,
-        "failed getting video info from caps %" GST_PTR_FORMAT, caps);
+      "failed getting video info from caps %" GST_PTR_FORMAT, (void *) caps);
     return FALSE;
   }
 }
@@ -779,7 +779,7 @@ static gboolean gst_nv_v4l2_camera_set_c
 
   GstNvV4l2CameraSrc *src = GST_NVV4L2CAMERASRC (base);
 
-  GST_DEBUG_OBJECT (src, "Received caps %" GST_PTR_FORMAT, caps);
+  GST_DEBUG_OBJECT (src, "Received caps %" GST_PTR_FORMAT, (void*) caps);
 
   if (!gst_video_info_from_caps (&info, caps))
   {
@@ -971,7 +971,7 @@ static gboolean gst_nv_v4l2_camera_set_c
   FD_SET(src->video_fd, &(src->read_set));
 
   /* Setting timeout value of select to 5, this can be configured as required */
-  src->tv = (struct timeval){ 0 };
+  memset(&src->tv, 0, sizeof(src->tv));
   src->tv.tv_sec = DEQUE_TIMEOUT;
 
   return TRUE;
