Index: TensorRT/plugin/bertQKVToContextPlugin/fused_multihead_attention/include/fused_multihead_attention.h
===================================================================
--- TensorRT.orig/plugin/bertQKVToContextPlugin/fused_multihead_attention/include/fused_multihead_attention.h
+++ TensorRT/plugin/bertQKVToContextPlugin/fused_multihead_attention/include/fused_multihead_attention.h
@@ -167,12 +167,7 @@ extern uint32_t cubin_fmha_v1_fp16_384_6
 extern uint32_t cubin_fmha_v1_fp16_128_64_sm90_cu_cubin_len;
 extern uint32_t cubin_fmha_v1_fp16_96_64_sm90_cu_cubin_len;
 extern uint32_t cubin_fmha_v1_fp16_64_64_sm90_cu_cubin_len;
-#if !(defined(ENABLE_SM72) || defined(ENABLE_SM75) || defined(ENABLE_SM80) || defined(ENABLE_SM86)                     \
-    || defined(ENABLE_SM87) || defined(ENABLE_SM89) || defined(ENABLE_SM90))
-// TRT-17573: Remove SM72 support from this file by factoring out the common logic required by the
-// V2 headers into a separate header.
-#error This file can only be included one of sm 72, 75, 80, 86, 87, 89, or 90 are defined.
-#endif
+
 static const struct FusedMultiHeadAttentionKernelMetaInfoV1
 {
     Data_type mDataType;
Index: TensorRT/plugin/bertQKVToContextPlugin/fused_multihead_attention_v2/include/fused_multihead_attention_v2.h
===================================================================
--- TensorRT.orig/plugin/bertQKVToContextPlugin/fused_multihead_attention_v2/include/fused_multihead_attention_v2.h
+++ TensorRT/plugin/bertQKVToContextPlugin/fused_multihead_attention_v2/include/fused_multihead_attention_v2.h
@@ -272,9 +272,6 @@ extern uint32_t cubin_fmha_v2_fp16_128_6
 extern uint32_t cubin_fmha_v2_fp16_96_64_sm90_cu_cubin_len;
 extern uint32_t cubin_fmha_v2_fp16_64_64_sm90_cu_cubin_len;
 
-#if !(defined(ENABLE_SM72) || defined(ENABLE_SM75) || defined(ENABLE_SM80) || defined(ENABLE_SM86) || defined(ENABLE_SM87) || defined(ENABLE_SM89) || defined(ENABLE_SM90))
-#error This file can only be included one of sm 72, 75, 80, 86, 87, 89, or 90 are defined.
-#endif
 static const struct FusedMultiHeadAttentionKernelMetaInfoV2
 {
     Data_type mDataType;
