From 1b1b7925b370fd7e13a02c06db7ae779ceea4a9b Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 20 Oct 2019 09:04:10 -0700
Subject: [PATCH] Fix skipuid arg usage for tx2 in odmsign.func

---
 bootloader/odmsign.func | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: Linux_for_Tegra/bootloader/odmsign.func
===================================================================
--- Linux_for_Tegra.orig/bootloader/odmsign.func
+++ Linux_for_Tegra/bootloader/odmsign.func
@@ -493,7 +493,11 @@ odmsign ()
 			BCTARGS="--mb1_bct mb1_bct_MB1_sigheader.bct.signed ";
 			sigbins BINSARGS "_sigheader" "signed" SIGNEDBINSARGS;
 			BINSARGS="--bins \"${SIGNEDBINSARGS}\" ";
-			SKIPUID="";
+			if [ "${CHIPID}" = "0x18" ]; then
+				SKIPUID=""
+			else
+				SKIPUID="--skipuid ";
+			fi
 		else
 			flashername="${flashername}.signed";
 			chext bctfilename ${bctfilename} "bct";
