From 075baecfd2d63e3076e71ca56cc55e6b6e6a7d1c Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Mon, 3 May 2021 07:20:13 -0700
Subject: [PATCH] Fix missing t186 boot partitions in l4t_bup_gen.func

Signed-off-by: Matt Madison <matt@madison.systems>

---
 bootloader/l4t_bup_gen.func | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/bootloader/l4t_bup_gen.func b/bootloader/l4t_bup_gen.func
index a2c5c84..d592fd8 100644
--- a/bootloader/l4t_bup_gen.func
+++ b/bootloader/l4t_bup_gen.func
@@ -33,6 +33,7 @@ ENTRY_LIST[nano_qspi_uboot]=""
 ENTRY_LIST[nano_qspi_bl]=""
 ENTRY_LIST[nano_qspi_kernel]=""
 ENTRY_LIST[bl_and_kernel]=""
+ENTRY_LIST[full_init]=""
 
 signed_ext="signed"
 signed_dir="signed"
@@ -196,9 +197,13 @@ _generate_bl_update_payload()
 				"${signed_dir}"/nvtboot_sigheader.bin.${signed_ext} mb2 2 0 $_common_spec; \
 				"${signed_dir}"/spe_sigheader.bin.${signed_ext} spe-fw 2 0 $_common_spec; \
 				"${signed_dir}"/adsp-fw_sigheader.bin.${signed_ext} adsp-fw 2 0 $_common_spec; \
-				"${signed_dir}"/badpage_sigheader.bin.${signed_ext} badpage-fw 2 0 $_common_spec; \
-				"${signed_dir}"/tos-trusty_sigheader.img.${signed_ext} secure-os 2 0 $_common_spec;"
+				"${signed_dir}"/tos-trusty_sigheader.img.${signed_ext} secure-os 2 0 $_common_spec; \
+				bmp.blob BMP 2 0 $_common_spec;"
 		ENTRY_LIST[kernel]=""${signed_dir}"/${kernel_image_base}_sigheader.img.${signed_ext} kernel 2 0 $_common_spec;"
+		ENTRY_LIST[full_init]=""${signed_dir}"/badpage_sigheader.bin.${signed_ext} badpage-fw 2 0 $_common_spec; \
+				"${signed_dir}"/eks_sigheader.img.${signed_ext} eks 2 0 $_common_spec; \
+				"${signed_dir}"/dram-ecc_sigheader.bin.${signed_ext} dram-ecc-fw 2 0 $_common_spec;"
+
 
 		if [ "${fuselevel}" == "fuselevel_nofuse" ]; then
 			ENTRY_LIST[bl]+=""${signed_dir}"/mb1_dev.bin.${signed_ext} mb1 2 1 $_common_spec; \
@@ -240,7 +245,8 @@ _generate_bl_update_payload()
 				"${signed_dir}"/bpmp_t194_sigheader.bin.${signed_ext} bpmp-fw 2 0 $_common_spec; \
 				"${signed_dir}"/eks_sigheader.img.${signed_ext} eks 2 0 $_common_spec; \
 				"${signed_dir}"/adsp-fw_sigheader.bin.${signed_ext} adsp-fw 2 0 $_common_spec; \
-				"${signed_dir}"/camera-rtcpu-rce_sigheader.img.${signed_ext} rce-fw 2 0 $_common_spec;"
+				"${signed_dir}"/camera-rtcpu-rce_sigheader.img.${signed_ext} rce-fw 2 0 $_common_spec; \
+				bmp.blob BMP 2 0 $_common_spec;"
 		ENTRY_LIST[kernel]=""${signed_dir}"/${kernel_image_base}_sigheader.img.${signed_ext} kernel 2 0 $_common_spec;"
 
 		if [ "${fuselevel}" == "fuselevel_nofuse" ]; then
@@ -325,6 +331,13 @@ _generate_bl_update_payload()
 		ENTRY_LIST[bl_and_kernel]="${ENTRY_LIST[bl]};${ENTRY_LIST[kernel]}"
 	fi
 
+	# OE4T: extra 'full_init' payload for TX2 that also includes partitions
+	# not normally included in the t18x bl_and_kernel payload for first-time
+	# initialization.
+	if [ "${tegra_soc}" == "t18x" ]; then
+		ENTRY_LIST[full_init]="${ENTRY_LIST[bl_and_kernel]};${ENTRY_LIST[full_init]}"
+	fi
+
 	for list_name in "${!ENTRY_LIST[@]}"; do
 		if [ "${ENTRY_LIST["${list_name}"]}" == "" ]; then
 			continue
