From 86bceea19f6463b94c3005d78f626ecf8cc15c0b Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Mon, 7 Jun 2021 04:22:40 -0700
Subject: [PATCH] Fix location of bsp_version file in l4t_bup_gen.func

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Matt Madison <matt@madison.systems>
---
 bootloader/l4t_bup_gen.func | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bootloader/l4t_bup_gen.func b/bootloader/l4t_bup_gen.func
index 6c7e176..1e820c7 100644
--- a/bootloader/l4t_bup_gen.func
+++ b/bootloader/l4t_bup_gen.func
@@ -648,13 +648,13 @@ function l4t_bup_gen()
 	export OUT=$(pwd);
 
 	# get the bsp version to fill the entry list version.
-	if [ -f "${LDK_DIR}/nv_tegra/bsp_version" ]; then
-		source "${LDK_DIR}/nv_tegra/bsp_version"
+	if [ -e "bsp_version" ]; then
+		source bsp_version
 		BSP_VER="${BSP_BRANCH}"
 		BSP_VER+="${BSP_MAJOR}"
 		BSP_VER+="${BSP_MINOR}"
 	else
-		echo "# Unknow release"
+		echo "# Unknown release"
 		echo "bootloader update payload creation failed"
 		exit 1
 	fi
