From 761b541dbd62558b51174be52655658add742ddc Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Wed, 28 Feb 2024 20:00:08 +0000
Subject: [PATCH] Update tegraflash_impl_t234.py

Upstream-Status: pending

Signed-off-by: Matt Madison <matt@madison.systems>
---
 bootloader/tegraflash_impl_t234.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: Linux_for_Tegra/bootloader/tegraflash_impl_t234.py
===================================================================
--- Linux_for_Tegra.orig/bootloader/tegraflash_impl_t234.py
+++ Linux_for_Tegra/bootloader/tegraflash_impl_t234.py
@@ -2085,7 +2085,7 @@ class TFlashT23x_Base(object):
             xml_tree = ElementTree.parse(file)
             mode = xml_tree.getroot().get('mode')
 
-            for file_nodes in xml_tree.getiterator('file'):
+            for file_nodes in xml_tree.iter('file'):
                 # To dinf encryptcompress node - Marcos
                 sbknode = file_nodes.find('sbk')
                 if sbknode is None:
