From 761b541dbd62558b51174be52655658add742ddc Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Wed, 28 Feb 2024 20:00:08 +0000
Subject: [PATCH] Update tegraflash_impl_t234.py

Signed-off-by: Matt Madison <matt@madison.systems>
---
 bootloader/tegraflash_impl_t234.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootloader/tegraflash_impl_t234.py b/bootloader/tegraflash_impl_t234.py
index ed0cb64..1b7c5ee 100644
--- a/bootloader/tegraflash_impl_t234.py
+++ b/bootloader/tegraflash_impl_t234.py
@@ -1943,7 +1943,7 @@ class TFlashT23x_Base(object):
             xml_tree = ElementTree.parse(file)
             mode = xml_tree.getroot().get('mode')
 
-            for file_nodes in xml_tree.getiterator('file'):
+            for file_nodes in xml_tree.iter('file'):
                 filename = file_nodes.get('name')
                 meta_blob_sz = int(file_nodes.get('meta_blob_size'))
                 enc_file = self.tegraflash_oem_enc(filename, bct_flag, meta_blob_sz)
-- 
2.34.1

