Index: Linux_for_Tegra/bootloader/l4t_bup_gen.func
===================================================================
--- Linux_for_Tegra.orig/bootloader/l4t_bup_gen.func
+++ Linux_for_Tegra/bootloader/l4t_bup_gen.func
@@ -154,7 +154,8 @@ _generate_bl_update_payload()
 				signed/cboot_t194_sigheader.bin.${signed_ext} cpu-bootloader 2 0 $_common_spec; \
 				signed/tos_t194_sigheader.img.${signed_ext} secure-os 2 0 $_common_spec; \
 				signed/bpmp_t194_sigheader.bin.${signed_ext} bpmp-fw 2 0 $_common_spec; \
-				signed/camera-rtcpu-rce_sigheader.bin.${signed_ext} rce-fw 2 0 $_common_spec"
+				signed/camera-rtcpu-rce_sigheader.bin.${signed_ext} rce-fw 2 0 $_common_spec; \
+				signed/boot_sigheader.img.${signed_ext} kernel 2 0 $_common_spec"
 
 		# add multiple binaries into BUP
 		if [ "${fuselevel}" = "fuselevel_nofuse" ]; then
@@ -183,9 +184,8 @@ _generate_bl_update_payload()
 		_add_multi_bins "bootloader-dtb" \
 			"$(ls multi_signed/*/${bootloader_dtb_base}_sigheader.dtb.${signed_ext})"
 
-		# TODO: use signed kernel and kernel-dtb after secure boot supported by cboot
-		ENTRY_LIST=$ENTRY_LIST"; signed/${lnx_image} kernel 2 0 $_common_spec "
-		_add_multi_bins "kernel-dtb" "$(ls multi_signed/*/${lnx_dtb})"
+		_add_multi_bins "kernel-dtb" \
+			"$(ls multi_signed/*/*_sigheader.dtb.${signed_ext} | grep -v "bpmp")"
 
 		_add_multi_bins "BCT" "$(ls multi_signed/*/br_bct*.bct)"
 		_add_multi_bins "MB1_BCT" \
