From 4dcda9f95a724623ce2813d73486324243dc2266 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sat, 12 Oct 2024 06:43:52 -0700
Subject: [PATCH] L4TLauncher: allow for empty/missing APPEND line in
 extlinux.conf

Signed-off-by: Matt Madison <matt@madison.systems>
---
 .../Application/L4TLauncher/L4TLauncher.c     | 30 ++++++++++---------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
index ab86f3e0..3be26f82 100644
--- a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
+++ b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
@@ -1422,21 +1422,23 @@ ExtLinuxBoot (
   CHAR8                      SWModule[] = "kernel";
   INTN                       FdtStatus;
 
-  // Process Args
-  ArgSize = StrSize (BootOption->BootArgs) + MAX_CBOOTARG_SIZE;
-  NewArgs = AllocateCopyPool (ArgSize, BootOption->BootArgs);
-  if (NewArgs == NULL) {
-    Status = EFI_OUT_OF_RESOURCES;
-    goto Exit;
-  }
+  // Process Args if present
+  if (BootOption->BootArgs != NULL) {
+    ArgSize = StrSize (BootOption->BootArgs) + MAX_CBOOTARG_SIZE;
+    NewArgs = AllocateCopyPool (ArgSize, BootOption->BootArgs);
+    if (NewArgs == NULL) {
+      Status = EFI_OUT_OF_RESOURCES;
+      goto Exit;
+    }
 
-  Status = gBS->LocateProtocol (&gAndroidBootImgProtocolGuid, NULL, (VOID **)&AndroidBootProtocol);
-  if (!EFI_ERROR (Status)) {
-    if (AndroidBootProtocol->AppendArgs != NULL) {
-      Status = AndroidBootProtocol->AppendArgs (NewArgs, ArgSize);
-      if (EFI_ERROR (Status)) {
-        ErrorPrint (L"%a: Failed to get platform addition arguments\r\n", __FUNCTION__);
-        goto Exit;
+    Status = gBS->LocateProtocol (&gAndroidBootImgProtocolGuid, NULL, (VOID **)&AndroidBootProtocol);
+    if (!EFI_ERROR (Status)) {
+      if (AndroidBootProtocol->AppendArgs != NULL) {
+        Status = AndroidBootProtocol->AppendArgs (NewArgs, ArgSize);
+        if (EFI_ERROR (Status)) {
+          ErrorPrint (L"%a: Failed to get platform addition arguments\r\n", __FUNCTION__);
+          goto Exit;
+        }
       }
     }
   }
-- 
2.43.0

