From 9e77fe4f3a34eda6d131e33f9a5f11fbaa34d423 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sat, 4 Feb 2023 06:31:24 -0800
Subject: [PATCH 2/3] L4TLauncher: allow for empty/missing APPEND line in extlinux.conf

Upstream-Status: Pending
Signed-off-by: Matt Madison <matt@madison.systems>
Signed-off-by: Jose Quaresma <jose.quaresma@foundries.io>
---
 .../Application/L4TLauncher/L4TLauncher.c     | 30 ++++++++++---------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
index 2e1849b8..ac53522d 100644
--- a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
+++ b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
@@ -1265,21 +1265,23 @@ ExtLinuxBoot (
   EFI_HANDLE                 KernelHandle      = NULL;
   EFI_LOADED_IMAGE_PROTOCOL  *ImageInfo;
 
-  // Process Args
-  ArgSize = StrSize (BootOption->BootArgs) + MAX_CBOOTARG_SIZE;
-  NewArgs = AllocateCopyPool (ArgSize, BootOption->BootArgs);
-  if (NewArgs == NULL) {
-    Status = EFI_OUT_OF_RESOURCES;
-    goto Exit;
-  }
+  // Process Args if present
+  if (BootOption->BootArgs != NULL) {
+    ArgSize = StrSize (BootOption->BootArgs) + MAX_CBOOTARG_SIZE;
+    NewArgs = AllocateCopyPool (ArgSize, BootOption->BootArgs);
+    if (NewArgs == NULL) {
+      Status = EFI_OUT_OF_RESOURCES;
+      goto Exit;
+    }
 
-  Status = gBS->LocateProtocol (&gAndroidBootImgProtocolGuid, NULL, (VOID **)&AndroidBootProtocol);
-  if (!EFI_ERROR (Status)) {
-    if (AndroidBootProtocol->AppendArgs != NULL) {
-      Status = AndroidBootProtocol->AppendArgs (NewArgs, ArgSize);
-      if (EFI_ERROR (Status)) {
-        ErrorPrint (L"%a: Failed to get platform addition arguments\r\n", __FUNCTION__);
-        goto Exit;
+    Status = gBS->LocateProtocol (&gAndroidBootImgProtocolGuid, NULL, (VOID **)&AndroidBootProtocol);
+    if (!EFI_ERROR (Status)) {
+      if (AndroidBootProtocol->AppendArgs != NULL) {
+        Status = AndroidBootProtocol->AppendArgs (NewArgs, ArgSize);
+        if (EFI_ERROR (Status)) {
+          ErrorPrint (L"%a: Failed to get platform addition arguments\r\n", __FUNCTION__);
+          goto Exit;
+        }
       }
     }
   }
-- 
2.34.1

