From cb273c18ec361c4e8cd8a627ed7f44d76c302f90 Mon Sep 17 00:00:00 2001
From: Matt Madison <3137284+madisongh@users.noreply.github.com>
Date: Tue, 10 Feb 2026 12:44:40 -0800
Subject: [PATCH 5/6] fix(L4TLauncher): allow for empty/missing APPEND line in
 extlinux.conf (#149)

Upstream-Status: Backport [https://github.com/NVIDIA/edk2-nvidia/pull/149]

Signed-off-by: Matt Madison <matt@madison.systems>
---
 .../Application/L4TLauncher/L4TLauncher.c     | 30 ++++++++++---------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
index 76fb920c..cb6d8a1c 100644
--- a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
+++ b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
@@ -1448,21 +1448,23 @@ ExtLinuxBoot (
   CHAR8                                 SWModule[] = "kernel";
   INTN                                  FdtStatus;
 
-  // Process Args
-  ArgSize = StrSize (BootOption->BootArgs) + MAX_CBOOTARG_SIZE;
-  NewArgs = AllocateCopyPool (ArgSize, BootOption->BootArgs);
-  if (NewArgs == NULL) {
-    Status = EFI_OUT_OF_RESOURCES;
-    goto Exit;
-  }
+  // Process Args if present
+  if (BootOption->BootArgs != NULL) {
+    ArgSize = StrSize (BootOption->BootArgs) + MAX_CBOOTARG_SIZE;
+    NewArgs = AllocateCopyPool (ArgSize, BootOption->BootArgs);
+    if (NewArgs == NULL) {
+      Status = EFI_OUT_OF_RESOURCES;
+      goto Exit;
+    }
 
-  Status = gBS->LocateProtocol (&gNVIDIAPlatformKernelArgsProtocolGuid, NULL, (VOID **)&KernelArgsProtocol);
-  if (!EFI_ERROR (Status)) {
-    if (KernelArgsProtocol->Append != NULL) {
-      Status = KernelArgsProtocol->Append (NewArgs, ArgSize);
-      if (EFI_ERROR (Status)) {
-        ErrorPrint (L"%a: Failed to get platform addition arguments\r\n", __FUNCTION__);
-        goto Exit;
+    Status = gBS->LocateProtocol (&gNVIDIAPlatformKernelArgsProtocolGuid, NULL, (VOID **)&KernelArgsProtocol);
+    if (!EFI_ERROR (Status)) {
+      if (KernelArgsProtocol->Append != NULL) {
+        Status = KernelArgsProtocol->Append (NewArgs, ArgSize);
+        if (EFI_ERROR (Status)) {
+          ErrorPrint (L"%a: Failed to get platform addition arguments\r\n", __FUNCTION__);
+          goto Exit;
+        }
       }
     }
   }
-- 
2.43.0

