From 940d7636357f974f4b7cdac8f2d26eda7b22957a Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Thu, 4 Sep 2025 04:41:16 -0700
Subject: [PATCH] PrePi: use GCC5 instead of GCC in build options

Upstream-Status: Pending
Signed-off-by: Matt Madison <matt@madison.systems>
---
 Silicon/NVIDIA/PrePi/PrePi.inf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Silicon/NVIDIA/PrePi/PrePi.inf b/Silicon/NVIDIA/PrePi/PrePi.inf
index 26fbb9ba..2006f8cb 100644
--- a/Silicon/NVIDIA/PrePi/PrePi.inf
+++ b/Silicon/NVIDIA/PrePi/PrePi.inf
@@ -92,5 +92,5 @@
   gNVIDIATokenSpaceGuid.PcdTegraCombinedUartRxMailbox
 
 [BuildOptions]
-  *_GCC_*_DLINK_FLAGS = -pie -Wl,-T,$(MODULE_DIR)/Scripts/PrePi-PIE.lds
+  *_GCC5_*_DLINK_FLAGS = -pie -Wl,-T,$(MODULE_DIR)/Scripts/PrePi-PIE.lds
   *_CLANGDWARF_*_DLINK_FLAGS = -pie -Wl,-T,$(MODULE_DIR)/Scripts/PrePi-PIE-clang.lds
-- 
2.43.0

