From 9739e20646640237167ab03ffe5b147920b6b6c7 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 16 May 2021 05:02:25 -0700
Subject: [PATCH] Fix tests/meson.build syntax error

Signed-off-by: Matt Madison <matt@madison.systems>
---
 tests/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/meson.build b/tests/meson.build
index 3509a57..a1c1a70 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -98,7 +98,7 @@ if with_glx
 
   foreach t : [['basic', ['-t', '1', '-i', '1'], env_glx],
                ['loop', ['-t', '1', '-i', '250'], env_glx],
-               ['threads', ['-t', '5', '-i', '20000'], [env_glx, 'LD_PRELOAD=libpthread.so.0']],
+               ['threads', ['-t', '5', '-i', '20000'], env_glx + ['LD_PRELOAD=libpthread.so.0']],
               ]
     test(
       'glxmakecurrent (@0@)'.format(t[0]),
