From f207b5c31868837bbf8fa36c46bd02980c169278 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Mon, 13 Dec 2021 05:47:45 -0800
Subject: [PATCH] Fix indentation in nvgstplayer.c


Upstream-Status: Pending
---
 nvgst_sample_apps/nvgstplayer-1.0/nvgstplayer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nvgst_sample_apps/nvgstplayer-1.0/nvgstplayer.c b/nvgst_sample_apps/nvgstplayer-1.0/nvgstplayer.c
index eb0b086..3b7e7e8 100644
--- a/nvgst_sample_apps/nvgstplayer-1.0/nvgstplayer.c
+++ b/nvgst_sample_apps/nvgstplayer-1.0/nvgstplayer.c
@@ -903,8 +903,8 @@ goto_next_track (gpointer data)
     destroy_current_track ();
   }
 
-if(app->stats)
-  stats_func(app->pfData.average_fps, app->pfData.frames_rendered, app->pfData.frames_dropped);
+  if(app->stats)
+    stats_func(app->pfData.average_fps, app->pfData.frames_rendered, app->pfData.frames_dropped);
 
   NVGST_INFO_MESSAGE_V ("uriCount: %d,  uriTotal: %d", (gint) app->uriCount,
       (gint) app->uriTotal);
