From 87242175b09e470c7066970f7dcf797feb05c622 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 31 Jan 2021 06:21:47 -0800
Subject: [PATCH] Update makefile for OE builds

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>

---
 Makefile | 29 ++++++++++++++++-------------
 1 file changed, 16 insertions(+), 13 deletions(-)

Index: gst-nvdrmvideosink/Makefile
===================================================================
--- gst-nvdrmvideosink.orig/Makefile
+++ gst-nvdrmvideosink/Makefile
@@ -27,47 +27,45 @@
 #
 ###############################################################################
 
+prefix ?= /usr
+exec_prefix ?= $(prefix)
+libdir ?= $(exec_prefix)/lib
+
 SO_NAME := libgstnvdrmvideosink.so
-DEST_DIR ?= /usr/lib/aarch64-linux-gnu/gstreamer-1.0
 
 SRCS := gstnvdrmvideosink.c \
 	util/drmutil.c \
 	util/vt_switch.c
 
-INCLUDES += -I/usr/lib/aarch64-linux-gnu/glib-2.0/include/ \
-	    -I./util/ \
-	    -I/usr/include/ \
-	    -I/usr/include/libdrm/ \
-	    -I../
-
-INCLUDES += -I/usr/src/jetson_multimedia_api/include/
-
 PKGS := glib-2.0 \
 	gstreamer-1.0 \
 	gstreamer-base-1.0 \
-	gstreamer-video-1.0
+	gstreamer-video-1.0 \
+	libdrm \
+	egl
 
 OBJS := $(SRCS:.c=.o)
 
-CFLAGS := `pkg-config --cflags $(PKGS)`
+CFLAGS += `pkg-config --cflags $(PKGS)`
 
-LDFLAGS = -Wl,--no-undefined -L/usr/lib/aarch64-linux-gnu/tegra
+LDFLAGS += -Wl,--no-undefined
 
-LIBS = -lnvbufsurface -lnvbufsurftransform -lEGL -ldrm
+LIBS = -lnvbufsurface -lnvbufsurftransform
 
 LIBS += `pkg-config --libs $(PKGS)`
 
 all: $(SO_NAME)
 
 %.o: %.c
-	$(CC) -c $< $(CFLAGS) $(INCLUDES) -o $@
+	$(CC) -c $< $(CFLAGS) -Iutil -o $@
 
 $(SO_NAME): $(OBJS)
 	$(CC) -shared -o $(SO_NAME) $(OBJS) $(LIBS) $(LDFLAGS)
 
 .PHONY: install
 install: $(SO_NAME)
-	cp -vp $(SO_NAME) $(DEST_DIR)
+	install -d $(DESTDIR)$(libdir)/gstreamer-1.0
+	install -m 0644 $(SO_NAME) $(DESTDIR)$(libdir)/gstreamer-1.0/
 
 .PHONY: clean
 clean:
